---
title: "Plots for manuscript: Systems epigenetic approach towards non-invasive breast cancer detection"
output: html_notebook
editor_options: 
  chunk_output_type: console
---

# Settings 

```{r setup}
library(patchwork)
library(ggtext)
library(ggplot2)
library(ggforce)
library(here)
library(ggpubr)
library(gtsummary)
library(dplyr)
library(gt)
library(tidyr)
library(karyoploteR)
library(epitools)
library(wesanderson)
library(openxlsx)


source(here("0-data","src", "plot_roc_dichot.R"))
source(here("0-data","src", "plot_triple_roc.R"))
source(here("0-data","src", "plot_double_roc.R"))
source(here("0-data","src", "plot_single_roc.R"))
source(here("0-data","src","plot_performance_comparisons.R"))
source(here("0-data","src", "manhattan.R"))
source(here("0-data","src", "coord_save_excel.R"))

outPath <- here("4-markdown-figures","pdf")
sepPath <- here("4-markdown-figures","separate-plots")
supPath <- here("5-supplements")
jpgPath <- here("4-markdown-figures","jpg")
coordPath <- here("4-markdown-figures","Source Data")

# dbPath <- fs::path_expand("~/Dropbox")
dbPath <- sub(" Dropbox.*", " Dropbox/Bente Theeuwes", getwd())
```


## Load dataframes

```{r load.df}
# Master dataframe
load(here("0-data","data.Rdata")) # data

# Training
load(here("0-data","dataframes-plotting","buccal-training.Rdata")) #pdat
training_buccal <- pdat
load(here("0-data","dataframes-plotting","cervical-training.Rdata")) #pdat
training_cervical <- pdat
load(here("0-data","dataframes-plotting","blood-training.Rdata")) #pdat
training_blood <- pdat

# Indices in validation data
load(here("0-data","dataframes-plotting","buccal_validation.Rdata")) # buccal_val
load(here("0-data","dataframes-plotting","cervical_validation.Rdata")) # cervical_val
load(here("0-data","dataframes-plotting","blood_validation.Rdata")) # blood_val
load(here("0-data","dataframes-plotting","brca_validation.Rdata")) # brca_val
load(here("0-data","dataframes-plotting","tissue_at_risk_validation.Rdata")) # tissue_at_risk_val
load(here("0-data","dataframes-plotting","GSE225845_validation.Rdata")) # GSE225845_val (breast tissue)
load(here("0-data","dataframes-plotting","tcga_validation.Rdata")) # tcga_val (breast tissue)
load(here("0-data","dataframes-plotting","3c_blood_validation.Rdata")) # buccal_val

# Principal components in all data and separate for all tissues
```

## Calculations for plotting

```{r plot.calc}
buccal_val$ic_stratification <- buccal_val$ic <= median(buccal_val$ic)
cervical_val$ic_stratification <- cervical_val$ic <= median(cervical_val$ic)
blood_val$neutro_stratification <- blood_val$hepidish_Neutro <= median(blood_val$hepidish_Neutro)
blood_3c_val$neutro_stratification <- blood_3c_val$hepidish_Neutro <= median(blood_3c_val$hepidish_Neutro)
```

## Order of levels, naming

```{r levels}
# Here we change the order of the levels, such that AUC calculation goes right.
buccal_val$type <- factor(buccal_val$type, levels = c('Control', 'BC case')) 
cervical_val$type <- factor(cervical_val$type, levels = c('Control', 'BC case'))
blood_val$type <- factor(blood_val$type, levels = c('Control', 'BC case'))
blood_3c_val$type <- factor(blood_3c_val$type, levels = c('Control', 'BC case'))
```

## Colours and text size

```{r cols}
pal.jama <- ggsci::pal_jama()(5)

# Two cases: Breast vs Control
colour.Breast <- pal.jama[4]
colour.Control <- pal.jama[3]

# Basic colours used for plotting
colour.green <- "#1BA68C"
colour.blue <- "#3B5C8D"
colour.red <- "#A61B35"
colour.breast <- "grey"
colour.sum <- "#D8C99E"

# Size of AUC label text
textSize = 2.1
```

## Plot functions

```{r}
pie_chart_dmps <- function(fileName, supPath=here("5-supplements")){
  colour.pal.d8 <- c("#EA7580","#F2949A","#F6B3A1","#D8C99E","#3AB9AC","#109DB7","#0C6EA5","#172869")
  colour.pal.c <- colorRampPalette(colour.pal.d8)
  
  # Read table from supplements folder
  df <- read.csv(file.path(supPath, fileName))
  
  # 1. Select entries with padj < 0.05
  selected_entries <- df[df$padj < 0.05, ]
  
  sets <- length(unique(selected_entries$Relation_to_Island))
  
  # 3. Create a pie chart of relation to island
  pie_plot <- ggplot(selected_entries, aes(x = "", fill = Relation_to_Island)) +
  geom_bar(stat = "count", width = 1) +
  coord_polar("y") +
  scale_fill_manual(values = colour.pal.d8) +  # Specify custom colors
  labs(fill = "Relation to Island", x = NULL, y = NULL) +
  theme_void()
  
  return(pie_plot)

}
```



# Table 1:  Discovery Sets


```{r tab1}
theme_gtsummary_journal(journal = "jama")
theme_gtsummary_compact()

tmp1 <- data |> 
  # Mutate
  dplyr::filter(set %in% c("discovery_buccal", "discovery_cervical", "discovery_blood")) |> 
  
  # Mutate for formatting
  dplyr::mutate(type = case_when(type == "BC case" ~ "BC",
                                 type == "Control" ~ "CO",
                                 TRUE ~ type),
                set = case_when(set == "discovery_buccal" ~ "Buccal<br>Discovery set",
                                    set == "discovery_cervical" ~ "Cervical<br>Discovery set",
                                    set == "discovery_blood" ~ "Blood<br>Discovery set"),
                set = factor(set, levels = c("Buccal<br>Discovery set",
                                                     "Cervical<br>Discovery set",
                                                     "Blood<br>Discovery set")))

# Columns select
tmp1 <- tmp1 |> select(set, age, type, BMI, menopause, smoking, stage_T, stage_N, grade, ER, PR)

# Replace empty columns with NA
tmp1 <- tmp1 |> mutate_if(is.character, ~na_if(., ''))

# Smoking
tmp1$smoking<- as.character(tmp1$smoking)
tmp1$smoking[tmp1$smoking == ""] <- "Unknown"
tmp1$smoking[is.na(tmp1$smoking)] <- "Unknown"

# Stage T
tmp1$stage_T[is.na(tmp1$stage_T) & tmp1$type=="BC"] <- "Unknown"
tmp1$stage_T[is.na(tmp1$stage_T)  & tmp1$type=="BC"] <- "Unknown"

# Stage N
tmp1$stage_N[is.na(tmp1$stage_N)  & tmp1$type=="BC"] <- "Unknown"
tmp1$stage_N[is.na(tmp1$stage_N)  & tmp1$type=="BC"] <- "Unknown"

tmp1$smoking <- factor(tmp1$smoking, levels = c("No", "Yes","Unknown"))

# grade
tmp1$grade[tmp1$type=="BC"  & is.na(tmp1$grade)] <- "Unknown"

# ER
tmp1$ER[tmp1$type=="BC" & is.na(tmp1$ER)] <- "Unknown"

# PR
tmp1$PR[tmp1$type=="BC"  & is.na(tmp1$PR)] <- "Unknown"

tmp1$stage_N[tmp1$stage_N %in% c("NX", "N")] <- "Unknown"

# Columns to convert to factors
columns_to_factor <- c("PR", "stage_T",   "stage_N",   "grade" ,    "ER")

# Convert selected columns to factors
tmp1[, columns_to_factor] <- lapply(tmp1[, columns_to_factor], factor)



tbl <- tmp1 |> 
  tbl_strata(strata = set,
             .tbl_fun= 
               ~ .x |> 
               tbl_summary(by = "type",
                           type = list("age" ~ "continuous",
                                       "BMI" ~ "continuous"),
                           statistic = list("age" ~ "{mean} ({sd})",
                                            "BMI" ~ "{mean} ({sd})"),
                           missing = "no",
                           label = list("age" = "Age at sample taken",
                                        "menopause" = "Menopausal status",
                                        "type" = "Type",
                                        "BMI" = "Body Mass Index",
                                        "smoking" = "Current smoker",
                                        "stage_T" = "Stage (T)",
                                        "stage_N" = "Stage (N)",
                                        "grade" = "Grade",
                                        "ER" = "ER",
                                        "PR" = "PR")) |> 
               bold_labels() |> 
               modify_header(all_stat_cols() ~"**{level}**<br>n = {n}")) 


t1 <- tbl |> 
  as_gt()  |> 
  text_transform(locations = cells_body(),
                 fn = function(x){
                   paste0(ifelse(x == "0 (0)", "-", ifelse(x == "NA (NA – NA)", "-", x)))
                 }
  ) |> 
  text_transform(locations = cells_body(),
                 fn = function(x){
                   paste0(ifelse(x == "0.0000 (0.0000–0.0000)", "0 (0–0)", x))
                 }
  ) |> 
  text_transform(locations = cells_body(),
                 fn = function(x){
                   paste0(ifelse(x == "0 (NA)", "-", x))
                 }
  ) |> 
  text_transform(locations = cells_body(),
                 fn = function(x){
                   paste0(ifelse(x == "NA (NA)", "-", x))
                 }
  ) |> 
  tab_options(
    table.width = px(550),
    table.font.size = 12,
    table.font.names = "Arial",
  ) |> 
  opt_stylize(style = 4)

t1 |> 
  gtsave(file.path(outPath,"T1.html"))

# to pdf
pagedown::chrome_print(file.path(outPath,"T1.html"),
                       output = file.path(outPath,"T1.pdf"))
```

# Table 2:  Validation Sets 

```{r}
theme_gtsummary_journal(journal = "jama")
theme_gtsummary_compact()

load(here("0-data", "data.Rdata")) # Load master dataframe
load(file.path(dataDir,"GSE237036","GSE237036_pheno_trimmed.Rdata")) # Load pheno blood

# Merge data frames by matching column names
data <- merge(data, pheno_trimmed, by = intersect(names(data), names(pheno_trimmed)), all.x = TRUE, all.y=TRUE)

tmp1 <- data |> 
  # Mutate
  dplyr::filter((set %in% c("validation_buccal", "validation_cervical")) | (subset == "GSE237036")) |> 
  
  # Mutate for formatting
  dplyr::mutate(type = case_when(type == "BC case" ~ "BC",
                                 type == "Control" ~ "CO",
                                 TRUE ~ type),
                set = case_when(set == "validation_buccal" ~ "Buccal<br>Validation set",
                                    set == "validation_cervical" ~ "Cervical<br>Validation set",
                                    set == "validation_blood" ~ "Blood<br>GSE237036"),
                set = factor(set, levels = c("Buccal<br>Validation set",
                                                     "Cervical<br>Validation set",
                                                     "Blood<br>GSE237036")))

# Columns select
tmp1 <- tmp1 |> select(set, age, type, BMI, menopause, smoking, stage_T, stage_N, grade, ER, PR)

# Replace empty columns with NA
tmp1 <- tmp1 |> mutate_if(is.character, ~na_if(., ''))

# Re-naming, making sure entries add up

# Smoking
tmp1$smoking<- as.character(tmp1$smoking)
tmp1$smoking[is.na(tmp1$smoking) == ""] <- "Unknown"
tmp1$smoking[is.na(tmp1$smoking) & tmp1$set=="Cervical<br>Validation set"] <- "Unknown"
tmp1$smoking[is.na(tmp1$smoking) & tmp1$set=="Buccal<br>Validation set"] <- "Unknown"

# Stage T
tmp1$stage_T[is.na(tmp1$stage_T) & tmp1$set=="Cervical<br>Validation set"  & tmp1$type=="BC"] <- "Unknown"
tmp1$stage_T[is.na(tmp1$stage_T) & tmp1$set=="Buccal<br>Validation set"  & tmp1$type=="BC"] <- "Unknown"

# Stage N
tmp1$stage_N[is.na(tmp1$stage_N) & tmp1$set=="Buccal<br>Validation set"  & tmp1$type=="BC"] <- "Unknown"
tmp1$stage_N[is.na(tmp1$stage_N) & tmp1$set=="Cervical<br>Validation set"  & tmp1$type=="BC"] <- "Unknown"

tmp1$smoking <- factor(tmp1$smoking, levels = c("No", "Yes","Unknown"))

# grade
tmp1$grade[tmp1$type=="BC" & tmp1$set=="Cervical<br>Validation set" & is.na(tmp1$grade)] <- "Unknown"

# ER
tmp1$ER[tmp1$type=="BC" & tmp1$set=="Cervical<br>Validation set" & is.na(tmp1$ER)] <- "Unknown"

# PR
tmp1$PR[tmp1$type=="BC" & tmp1$set=="Cervical<br>Validation set" & is.na(tmp1$PR)] <- "Unknown"

# Columns to convert to factors
columns_to_factor <- c("PR", "stage_T",   "stage_N",   "grade" ,    "ER")

# Convert selected columns to factors
tmp1[, columns_to_factor] <- lapply(tmp1[, columns_to_factor], factor)




tbl <- tmp1 |>
  tbl_strata(strata = set,
             .tbl_fun= 
               ~ .x |>
               tbl_summary(by = type,
                           type = list(age ~ "continuous",BMI ~ "continuous"),
                           statistic = list(age ~ "{mean} ({sd})",BMI ~ "{mean} ({sd})"),
                           missing = "no",
                           label = list(age = "Age at sample taken",
                                        menopause = "Menopausal status",
                                        type = "Type",
                                        BMI = "Body Mass Index",
                                        smoking = "Current smoker",
                                        stage_T = "Stage (T)",
                                        stage_N = "Stage (N)",
                                        grade = "Grade",
                                        ER = "ER",
                                        PR = "PR")) |>
               bold_labels() |>
               modify_header(all_stat_cols() ~"**{level}**<br>n = {n}")) 


t1 <- tbl |>
  as_gt()  |>
  text_transform(locations = cells_body(),
                 fn = function(x){
                   paste0(ifelse(x == "0 (0)", "-", ifelse(x == "NA (NA – NA)", "-", x)))
                 }
  ) |>
  text_transform(locations = cells_body(),
                 fn = function(x){
                   paste0(ifelse(x == "0.0000 (0.0000–0.0000)", "0 (0–0)", x))
                 }
  ) |>
  text_transform(locations = cells_body(),
                 fn = function(x){
                   paste0(ifelse(x == "0 (NA)", "-", x))
                 }
  ) |>
  text_transform(locations = cells_body(),
                 fn = function(x){
                   paste0(ifelse(x == "NA (NA)", "-", x))
                 }
  ) |>
  tab_options(
    table.width = px(550),
    table.font.size = 12,
    table.font.names = "Arial",
  ) |> 
  opt_stylize(style = 4)

t1 |> gtsave(file.path(outPath,"T2.html"))

# to pdf
pagedown::chrome_print(file.path(outPath,"T2.html"), output = file.path(outPath,"T2.pdf"))
```

# Figure 1

**Note that the final ordering of this figure was done in Adobe Illustrator, hence do not regenerate this figure entirely whenever changes need to be made. Please open 'F1.ai' and change there.**

## a) overview diagram

```{r}
# Import the image
img <- jpeg::readJPEG(file.path(sepPath,"1a.jpg"), native = TRUE) # Created with Adobe Illustrator 

ratio = 600 /1600

a1 <- ggplot() +
  background_image(img) + theme(aspect.ratio = ratio) +coord_fixed()
```


## b) Grid with hexbin / eFORGE / mean methylation

```{r}
source(file.path(here(),"4-markdown-figures","fig1b_code_chunks.R"))

# Save coordinates
b1 <- BHo_CHer$plot
c1 <- Breast_BHo_CHer
d1 <- Breast_BHer_CHer
e1 <- overall
f1 <- BHo_CHo$plot
g1 <- Breast_BHo_CHo
h1 <- BHer_CHo$plot
i1 <- Breast_BHer_CHo

coord_save_excel(b1, coordPath)
coord_save_excel(c1, coordPath)
coord_save_excel(d1, coordPath)
coord_save_excel(e1, coordPath)
coord_save_excel(f1, coordPath)
coord_save_excel(g1, coordPath)
coord_save_excel(h1, coordPath)
coord_save_excel(i1, coordPath)

```


## Compile and save

```{r, message=FALSE, echo=FALSE}

layout <- "
####
####
ABCD
EEEE
EEEE
FGHI
"


pdf_width <- 8.3  # Width in inches (a4)
pdf_height <- 11.7   # Height in inches (a4: 11.7)


compiled <-  BHo_CHer$plot + Breast_BHo_CHer + guide_area() + Breast_BHer_CHer  + wrap_elements(overall) + BHo_CHo$plot +  Breast_BHo_CHo + BHer_CHo$plot + Breast_BHer_CHo + plot_layout(guides = 'collect')
  
  
compiled <-  compiled + plot_annotation(tag_levels = 'a') + 
  plot_layout(design = layout)

ggsave(file.path(outPath,"F1_new.pdf"), width = pdf_width, height = pdf_height, compiled)
ggsave(file.path(jpgPath,"F1_new.jpg"), width = pdf_width, height = pdf_height, compiled)
```


# Figure 2

## a) WID-buccal versus ic

```{r}
a2 <- ggplot(data = buccal_val, aes(x = ic, y = WID_buccal_BC, color = type, fill = factor(type))) +
  geom_point(alpha=0.4,size=1) +  # Plot points
  geom_smooth(method = "lm", se = TRUE, aes(group = type)) +
  scale_color_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control)) +  # Add linear fit with confidence interval
  labs(
    y = "WID-buccal-BC index",
    x = "immune cell proportion"
  ) +
  theme_minimal() +
  theme(legend.title = element_blank()) +
  theme(legend.position = "none") +  # Remove legend
  theme(panel.background = element_blank()) + 
  stat_cor(method = "pearson", show.legend = FALSE, label.x.npc='left', label.y.npc='top', size = 3) + 
  scale_fill_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control))

a2


# Save coordinates
coord_save_excel(a2, coordPath)

```

## b) WID-buccal versus age

```{r}

b2 <- ggplot(data = buccal_val, aes(x = age, y = WID_buccal_BC, color = type, fill = factor(type))) +
  geom_point(alpha=0.4,size=1) +  # Plot points
  geom_smooth(method = "lm", se = TRUE, aes(group = type)) +
  scale_color_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control)) +  # Add linear fit with confidence interval
  labs(
    y = "WID-buccal-BC index",
    x = "Subject age"
  ) +
  theme_minimal() +
  theme(legend.title = element_blank()) +
  theme(legend.position = "none") +  # Remove legend
  theme(panel.background = element_blank()) + 
  stat_cor(method = "pearson", show.legend = FALSE, label.x.npc='left', label.y.npc='top', size = 3) + 
  scale_fill_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control))

b2

# Save coordinates
coord_save_excel(b2, coordPath)

```

## c) Boxplot WID-buccal (ic and age-adjusted)

```{r}

c2 <- ggplot(buccal_val,
             aes(x = factor(type), y = WID_buccal_BC_adj)) +
  geom_boxplot(aes(fill = type),
               alpha = 0.2) +
  ggbeeswarm::geom_quasirandom(aes(colour = type),
                               alpha = 0.5, size = 0.5) +
  scale_color_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control)) +  # Add linear fit with confidence interval
  labs(
    x = '',
    y = "WID-buccal-BC index (adjusted)"
  ) +
  theme_minimal() +
  theme(legend.title = element_blank()) +
  theme(legend.position = "none",axis.text.x = element_text(angle = 45, hjust = 1)) +  # Remove legend
  theme(panel.background = element_blank()) + 
  ggpubr::stat_compare_means(comparisons = list(c('Control', 'BC case')), label = 'p.format',vjust = 0, size=2) +
  scale_fill_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control))

c2

# Save coordinates
coord_save_excel(c2, coordPath)

```

## d) WID-buccal AUC

```{r}
# Assuming buccal_val is your data frame
less <- buccal_val |>
  filter(ic_stratification == TRUE)
more <- buccal_val |>
  filter(ic_stratification == F)


d2 <- plot_triple_roc(factor(buccal_val$type), buccal_val$WID_buccal_BC_adj, factor(less$type), less$WID_buccal_BC_adj, factor(more$type), more$WID_buccal_BC_adj, col1 = colour.green, col2 = colour.blue, col3 = colour.sum, title1 = "All", title2 = paste("IC<", round(median(buccal_val$ic), 2), sep = ""), title3 = paste("IC>", round(median(buccal_val$ic), 2), sep = ""), style = "default",
                            direction1 = "<",direction2 = "<", generalTitle="", textCol1 = "black",textCol3="black",textSize=2.1)

d2

# Save coordinates
coord_save_excel(d2, coordPath)
```

## e) WID-cervical versus ic

```{r}

e2 <- ggplot(data = cervical_val, aes(x = ic, y = WID_cervical_BC, color = type, fill = factor(type))) +
  geom_point(alpha=0.4,size=1) +  # Plot points
  geom_smooth(method = "lm", se = TRUE, aes(group = type)) +
  scale_color_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control)) +  # Add linear fit with confidence interval
  labs(
    y = "WID-cervical-BC index",
    x = "immune cell proportion"
  ) +
  theme_minimal() +
  theme(legend.title = element_blank()) +
  theme(legend.position = "none") +  # Remove legend
  theme(panel.background = element_blank()) + 
  stat_cor(method = "pearson", show.legend = FALSE, label.x.npc='left', label.y.npc='top', size = 3) + 
  scale_fill_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control))

e2

# Save coordinates
coord_save_excel(e2, coordPath)

```

## f) WID-cervical versus age

```{r}
f2 <- ggplot(data = cervical_val, aes(x = age, y = WID_cervical_BC, color = type, fill = factor(type))) +
  geom_point(alpha=0.4,size=1) +  # Plot points
  geom_smooth(method = "lm", se = TRUE, aes(group = type)) +
  scale_color_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control)) +  # Add linear fit with confidence interval
  labs(
    y = "WID-cervical-BC index",
    x = "Subject age"
  ) +
  theme_minimal() +
  theme(legend.title = element_blank()) +
  theme(legend.position = "none") +  # Remove legend
  theme(panel.background = element_blank()) + 
  stat_cor(method = "pearson", show.legend = FALSE, label.x.npc='left', label.y.npc='top', size = 3) + 
  scale_fill_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control))

f2

# Save coordinates
coord_save_excel(f2, coordPath)

```

## g) Boxplot WID-cervical (ic and age-adjusted)

```{r}
g2 <- ggplot(cervical_val,
             aes(x = factor(type), y = WID_cervical_BC_adj)) +
  geom_boxplot(aes(fill = type),
               alpha = 0.2) +
  ggbeeswarm::geom_quasirandom(aes(colour = type),
                               alpha = 0.5, size = 0.5) +
  scale_color_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control)) +  # Add linear fit with confidence interval
  labs(
    x = '',
    y = "WID-cervical-BC index (adjusted)"
  ) +
  theme_minimal() +
  theme(legend.title = element_blank()) +
  theme(legend.position = "none",axis.text.x = element_text(angle = 45, hjust = 1)) +  # Remove legend
  theme(panel.background = element_blank()) + 
  ggpubr::stat_compare_means(comparisons = list(c('Control', 'BC case')), label = 'p.format',vjust = 0, size=2) +
  scale_fill_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control))

g2

# Save coordinates
coord_save_excel(g2, coordPath)
```

## h) WID-cervical AUC

```{r}
less <- cervical_val |>
  filter(ic_stratification == TRUE)
more <- cervical_val |>
  filter(ic_stratification == F)


h2 <- plot_triple_roc(factor(cervical_val$type), cervical_val$WID_cervical_BC_adj, factor(less$type), less$WID_cervical_BC_adj, factor(more$type), more$WID_cervical_BC_adj, col1 = colour.green, col2 = colour.blue, col3 = colour.sum, title1 = "All", title2 = paste("IC<", round(median(cervical_val$ic), 2), sep = ""), title3 = paste("IC>", round(median(cervical_val$ic), 2), sep = ""), style = "default",
                            direction1 = "<",direction2 = "<", generalTitle="", textCol1 = "black",textCol3="black",textSize=2.1)

h2

# Save coordinates
coord_save_excel(h2, coordPath)
```


## i) WID-blood versus neutro

```{r}
i2 <- ggplot(data = blood_val, aes(x = hepidish_Neutro, y = WID_blood_BC, color = type, fill = factor(type))) +
  geom_point(alpha=0.4,size=1) +  # Plot points
  geom_smooth(method = "lm", se = TRUE, aes(group = type)) +
  scale_color_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control)) +  # Add linear fit with confidence interval
  labs(
    y = "WID-blood-BC index",
    x = "neutrophil proportion"
  ) +
  theme_minimal() +
  theme(legend.title = element_blank()) +
  theme(legend.position = "top") +  # Remove legend
  theme(panel.background = element_blank()) + 
  stat_cor(method = "pearson", show.legend = FALSE, label.x.npc='left', label.y.npc='top', size = 3) + 
  scale_fill_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control)) +  
  guides(color = guide_colorbar(title = "Horsepower"),
    shape = guide_legend(title = "Weight"), size = guide_legend(title = "Gear")
  )

i2

# Save coordinates
coord_save_excel(i2, coordPath)
```

## j) Boxplot WID-blood (neutro-adjusted)

```{r}
j2 <- ggplot(blood_val,
             aes(x = factor(type), y = WID_blood_BC_adj)) +
  geom_boxplot(aes(fill = type),
               alpha = 0.2) +
  ggbeeswarm::geom_quasirandom(aes(colour = type),
                               alpha = 0.5, size = 0.5) +
  scale_color_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control)) +  # Add linear fit with confidence interval
  labs(
    x = '',
    y = "WID-blood-BC index (adjusted)"
  ) +
  theme_minimal() +
  theme(legend.title = element_blank()) +
  theme(legend.position = "top",axis.text.x = element_text(angle = 45, hjust = 1)) +  # Remove legend
  theme(panel.background = element_blank()) + 
  # ggpubr::stat_compare_means(comparisons = list(c('Control', 'BC case')), label = 'p.signif',vjust = 0.3) +
  scale_fill_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control))

j2

# P value
# wilcox.test(blood_val[blood_val$type=='Control',]$WID_blood_BC_adj, blood_val[blood_val$type!='Control',]$WID_blood_BC_adj)

# Save coordinates
coord_save_excel(j2, coordPath)
```

## k) WID-blood AUC

```{r}
less <- blood_val |>
  filter(neutro_stratification == TRUE)
more <- blood_val |>
  filter(neutro_stratification == F)

k2 <- plot_triple_roc(factor(blood_val$type), blood_val$WID_blood_BC_adj, factor(less$type), less$WID_blood_BC_adj, factor(more$type), more$WID_blood_BC_adj, col1 = colour.green, col2 = colour.blue, col3 = colour.sum, title1 = "All", title2 = paste("Neutro<", round(median(blood_val$hepidish_Neutro), 2), sep = ""), title3 = paste("Neutro>", round(median(blood_val$hepidish_Neutro), 2), sep = ""), style = "default",
                            direction1 = "<",direction2 = "<", generalTitle="", textCol1 = "black",textCol3="black",textSize=2.1)

k2

# Save coordinates
coord_save_excel(k2, coordPath)
```


## Compile and save

```{r, message=FALSE, echo=FALSE}

layout <- "
AAABBBCCDDD
EEEFFFGGHHH
IIIJJJKKLLL
"


pdf_width <- 8.3  # Width in inches (a4)
pdf_height <- 9   # Height in inches (a4: 11.7)


compiled <- a2 + b2 + c2 + d2 + e2 + f2 + g2 + h2 + guide_area() + i2 + j2 + k2 
  
compiled <-  compiled + plot_annotation(tag_levels = 'a') + 
  plot_layout(design = layout, guides = 'collect')

ggsave(file.path(outPath,"F2.pdf"), width = pdf_width, height = pdf_height, compiled) # Manually fix legend in illustrator
# ggsave(file.path(jpgPath,"F2.jpg"), width = pdf_width, height = pdf_height, compiled)
```

# Figure 3

Non-significant comparisons are commented out for cleanliness of the figures. Each section/figure below contains the original code to check significance.

## a) WID-buccal: association with stage

```{r}
a3 <- buccal_val |> 
  dplyr::mutate(stage = ifelse(type == 'Control', 'Control', stage_T)) |> 
  dplyr::filter(stage != '') |> 
  ggplot(aes(x = stage, y = WID_buccal_BC_adj)) +
  geom_boxplot(aes(fill = type),
               alpha = 0.2,outlier.shape = NA) +
  ggbeeswarm::geom_quasirandom(aes(colour = type),
                               alpha = 0.5, size = 0.5) +
  scale_color_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control)) +  # Add linear fit with confidence interval
  labs(
    x = '',
    y = "WID-buccal-BC index\nadjusted for age and ic") +
  theme_minimal() +
  theme(legend.title = element_blank()) +
  theme(legend.position = "top") +  # Remove legend
  theme(panel.background = element_blank()) + 
  ggpubr::stat_compare_means(comparisons = list(
    c('Control', 'T1'),
    c('Control', 'T2')
    # c('Control', 'T3'),
    # c('T1', 'T2'),
    # c('T1', 'T3'),
    # c('T2', 'T3')
    ),label = 'p.format', label.y = c(1.6),step.increase = 0.13, vjust = 0,textsize=1)  +
  scale_fill_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control)) +
   scale_y_continuous(expand = expansion(mult = c(0, 0.1)))


a3

# Save coordinates
coord_save_excel(a3, coordPath)
```


## b) WID-buccal: association with grade 


```{r}
b3 <- buccal_val |> 
  dplyr::mutate(grade = ifelse(type == 'Control', 'Control', gsub("Grade ", "", grade)))  |> 
  dplyr::filter(grade != 'Unknown') |> 
  ggplot(aes(x = grade,
             y = WID_buccal_BC_adj)) +
  geom_boxplot(aes(fill = type),
               alpha = 0.2,outlier.shape = NA) +
  ggbeeswarm::geom_quasirandom(aes(colour = type),
                               alpha = 0.5, size = 0.5) +
  scale_color_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control)) +  # Add linear fit with confidence interval
  labs(
    x = '',
    y = ""
  ) +
  theme_minimal() +
  theme(legend.title = element_blank()) +
  theme(legend.position = "none") +  # Remove legend
  theme(panel.background = element_blank()) + 
  ggpubr::stat_compare_means(comparisons = list(
    c('Control', 'II'),
#    c('Control', 'I'),
    c('Control', 'III')
#    c('I', 'II'),
#    c('I', 'III'),
#    c('II', 'III')
    ),label = 'p.format', label.y = c(1.7),step.increase = 0.13, vjust = 0,textsize=1)  +
  scale_fill_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control)) +
   scale_y_continuous(expand = expansion(mult = c(0, 0.1)))

b3

# Save coordinates
coord_save_excel(b3, coordPath)
```


## c) WID-buccal: Association with ER status 


```{r}


c3 <- buccal_val |> 
  dplyr::mutate(ERstat = ifelse(type == 'Control', 'Control', ifelse(ER == 'Pos', "ER+", "ER-")),
                ERstat = factor(ERstat, levels = c("Control", "ER+", "ER-")))  |> 
  dplyr::filter(ERstat != 'Unknown') |> 
  ggplot(aes(x = ERstat,
             y = WID_buccal_BC_adj)) +
  geom_boxplot(aes(fill = type),
               alpha = 0.2,outlier.shape = NA) +
  ggbeeswarm::geom_quasirandom(aes(colour = type),
                               alpha = 0.5, size = 0.5) +
  scale_color_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control)) +  # Add linear fit with confidence interval
  labs(
    x = '',
    y = ""
  ) +
  theme_minimal() +
  theme(legend.title = element_blank()) +
  theme(legend.position = "none") +  # Remove legend
  theme(panel.background = element_blank()) + 
  ggpubr::stat_compare_means(comparisons = list(
    c('Control', 'ER+'),
     c('Control', 'ER-')
  #  c('ER+', 'ER-')
    ),label = 'p.format', label.y = c(1.7),step.increase = 0.13, vjust = 0,textsize=1)  +
  scale_fill_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control))+
   scale_y_continuous(expand = expansion(mult = c(0, 0.1)))

c3

# Save coordinates
coord_save_excel(c3, coordPath)
```

## d) WID-buccal: Association with PR status 


```{r}


d3 <- buccal_val |> 
  dplyr::mutate(PRstat = ifelse(type == 'Control', 'Control', ifelse(PR == 'Pos', 'PR+', 'PR-')),
                PRstat = factor(PRstat, levels = c("Control", "PR+", "PR-")))  |> 
  dplyr::filter(PRstat != 'Unknown') |> 
  ggplot(aes(x = PRstat,
             y = WID_buccal_BC_adj)) +
  geom_boxplot(aes(fill = type),
               alpha = 0.2,outlier.shape = NA) +
  ggbeeswarm::geom_quasirandom(aes(colour = type),
                               alpha = 0.5, size = 0.5) +
  scale_color_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control)) +  # Add linear fit with confidence interval
  labs(
    x = '',
    y = ""
  ) +
  theme_minimal() +
  theme(legend.title = element_blank()) +
  theme(legend.position = "none") +  # Remove legend
  theme(panel.background = element_blank()) + 
  ggpubr::stat_compare_means(comparisons = list(
    c('Control', 'PR+'),
     c('Control', 'PR-')
    #c('PR+', 'PR-')
    ),label = 'p.format', label.y = c(1.7),step.increase = 0.13, vjust = 0,textsize=1)  +
  scale_fill_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control))+
   scale_y_continuous(expand = expansion(mult = c(0, 0.1)))

d3

# Save coordinates
coord_save_excel(d3, coordPath)
```

## e) WID-buccal: Association with HER2 status 


```{r}


e3 <- buccal_val |> 
  dplyr::mutate(HERstat = ifelse(type == 'Control', 'Control',
                                 ifelse(HER2 == 'Pos', 'HER2+', "HER2-")),
                HERstat = factor(HERstat, levels = c("Control", "HER2-", "HER2+")))  |> 
  dplyr::filter(HERstat != 'Unknown') |> 
  ggplot(aes(x = HERstat,
             y = WID_buccal_BC_adj)) +
  geom_boxplot(aes(fill = type),
               alpha = 0.2,outlier.shape = NA) +
  ggbeeswarm::geom_quasirandom(aes(colour = type),
                               alpha = 0.5, size = 0.5) +
  scale_color_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control)) +  # Add linear fit with confidence interval
  labs(
    x = '',
    y = "WID-buccal-BC index\nadjusted for age and ic") +
  theme_minimal() +
  theme(legend.title = element_blank()) +
  theme(legend.position = "none", axis.text.x = element_text(angle = 45, hjust = 1)) +  # Remove legend
  theme(panel.background = element_blank()) + 
  ggpubr::stat_compare_means(comparisons = list(
    c('Control', 'HER2-'),
     c('Control', 'HER2+')
  #  c('HER2+', 'HER2-')
    ),label = 'p.format', label.y = c(1.7),step.increase = 0.13, vjust = 0,textsize=1)  +
  scale_fill_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control))+
   scale_y_continuous(expand = expansion(mult = c(0, 0.1)))

e3

# Save coordinates
coord_save_excel(e3, coordPath)
```

## f) WID-buccal: PRS with WID-BC-Buccal

```{r}


f3 <- ggplot(buccal_val, aes(x = polygenic_risk_score_bc, y = WID_buccal_BC_adj, color = type, fill = factor(type))) +
  geom_point(alpha=0.4,size=1) +  # Plot points
  geom_smooth(method = "lm", se = TRUE, aes(group = type)) +
  scale_color_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control)) +  # Add linear fit with confidence interval
  labs(
    y = "",
    x = "Polygenic Risk \n Score") + theme_minimal() +
  theme(legend.title = element_blank()) +
  theme(legend.position = "top") +  # Remove legend
  theme(panel.background = element_blank()) +  
  stat_cor(method = "pearson", show.legend = FALSE, label.x.npc='left', label.y.npc='top', size = 2.8) + 
  scale_fill_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control)) 
f3


# Save coordinates
coord_save_excel(f3, coordPath)  

```

## g) WID-buccal: association with BRCAmut status

```{r}
brca_val$type <- factor(brca_val$type, levels = c('Control', 'BRCA1','BRCA2')) # Here we change the order of the levels

brca_buccal <- droplevels(brca_val[brca_val$sampletype=='buccal',])

# 1: Boxplot of WID-BC-buccal
g3 <- brca_buccal |> 
  ggplot(aes(x = factor(type), y = WID_buccal_BC_adj))  +
  geom_boxplot(aes(fill = type),
               alpha = 0.3,outlier.shape = NA) +
  ggbeeswarm::geom_quasirandom(aes(colour = type),
                               alpha = 0.5, size = 0.5) +  # Add beeswarm plot
  labs(x = "", y = "") +
  theme_minimal() +
  theme(legend.position = "none", axis.text.x = element_text(angle = 45, hjust = 1)) +
  ggpubr::stat_compare_means(comparisons = list(
    # c("Control", "BRCA1"),
    # c("Control", "BRCA2"),
    # c("BRCA1", "BRCA2")
  ), label = 'p.signif') +
  guides(fill = guide_legend(title.position = "top", title.hjust = 0.5)) + 
  scale_fill_manual(values = c("Control" = colour.Control, "BRCA1" = pal.jama[5], "BRCA2" = pal.jama[5]),
                    aesthetics = c('colour', 'fill')) 


g3
       
# Save coordinates
coord_save_excel(g3, coordPath)  
```

## h) WID-buccal: AUC of WID-BC-buccal in BRCA1/BRCA2 mutation carriers


```{r}
filtered_BRCA2 <- brca_buccal |>
  filter(type == 'Control' | type == 'BRCA2')
filtered_BRCA1 <- brca_buccal |>
  filter(type == 'Control' | type == 'BRCA1')

# Here we change the order of the levels, such that AUC calculation goes right.
filtered_BRCA2$type <- factor(filtered_BRCA2$type, levels = c('Control', 'BRCA2')) 
filtered_BRCA1$type <- factor(filtered_BRCA1$type, levels = c('Control', 'BRCA1')) 


h3 <- plot_double_roc(factor(filtered_BRCA1$type), filtered_BRCA1$WID_buccal_BC_adj, factor(filtered_BRCA2$type), filtered_BRCA2$WID_buccal_BC_adj, col1 = colour.blue, col2 = colour.green, title1 = "BRCA1", title2 = "BRCA2", style = "default",
                            direction1 = "<",direction2 = "<", generalTitle="",textSize = 2.1) 

h3

# Save coordinates
coord_save_excel(h3, coordPath)  
```

## i) WID-cervical: association with stage

```{r}



i3 <- cervical_val |> 
  dplyr::mutate(stage = ifelse(type == 'Control', 'Control', stage_T)) |> 
  dplyr::filter(stage != '') |> 
  ggplot(aes(x = stage, y = WID_cervical_BC_adj)) +
  geom_boxplot(aes(fill = type),
               alpha = 0.2,outlier.shape = NA) +
  ggbeeswarm::geom_quasirandom(aes(colour = type),
                               alpha = 0.5, size = 0.5) +
  scale_color_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control)) +  # Add linear fit with confidence interval
  labs(
    x = '',
    y = "WID-cervical-BC index\nadjusted for age and ic") +
  theme_minimal() +
  theme(legend.title = element_blank()) +
  theme(legend.position = "none") +  # Remove legend
  theme(panel.background = element_blank()) + 
  ggpubr::stat_compare_means(comparisons = list(
    c('Control', 'T1'),
     c('Control', 'T2')
    #  c('Control', 'T3'),
    # c('T1', 'T2'),
    # c('T1', 'T3'),
    # c('T2', 'T3')
    ),label = 'p.format', label.y = c(1.4),step.increase = 0.13, vjust = 0,textsize=1)  +
  scale_fill_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control)) +
   scale_y_continuous(expand = expansion(mult = c(0, 0.1)))


i3

# Save coordinates
coord_save_excel(i3, coordPath)  
```


## j) WID-cervical: association with grade 


```{r}


j3 <- cervical_val |> 
  dplyr::mutate(grade = ifelse(type == 'Control', 'Control', gsub("Grade ", "", grade)))  |> 
  dplyr::filter(grade != 'Unknown') |> 
  ggplot(aes(x = grade,
             y = WID_cervical_BC_adj)) +
  geom_boxplot(aes(fill = type),
               alpha = 0.2,outlier.shape = NA) +
  ggbeeswarm::geom_quasirandom(aes(colour = type),
                               alpha = 0.5, size = 0.5) +
  scale_color_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control)) +  # Add linear fit with confidence interval
  labs(
    x = '',
    y = ""
  ) +
  theme_minimal() +
  theme(legend.title = element_blank()) +
  theme(legend.position = "none") +  # Remove legend
  theme(panel.background = element_blank()) + 
  ggpubr::stat_compare_means(comparisons = list(
    c('Control', 'II'),
  #  c('Control', 'I'),
    c('Control', 'III')
  #  c('I', 'II'),
  #  c('I', 'III'),
 #   c('II', 'III')
    ),label = 'p.format', label.y = c(1.6),step.increase = 0.13, vjust = 0,textsize=1)  +
  scale_fill_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control)) +
   scale_y_continuous(expand = expansion(mult = c(0, 0.1)))

j3

# Save coordinates
coord_save_excel(j3, coordPath)  
```


## k) WID-cervical: Association with ER status 


```{r}


k3 <- cervical_val |> 
  dplyr::mutate(ERstat = ifelse(type == 'Control', 'Control', ifelse(ER == 'Pos', "ER+", "ER-")),
                ERstat = factor(ERstat, levels = c("Control", "ER+", "ER-")))  |> 
  dplyr::filter(ERstat != 'Unknown') |> 
  ggplot(aes(x = ERstat,
             y = WID_cervical_BC_adj)) +
  geom_boxplot(aes(fill = type),
               alpha = 0.2,outlier.shape = NA) +
  ggbeeswarm::geom_quasirandom(aes(colour = type),
                               alpha = 0.5, size = 0.5) +
  scale_color_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control)) +  # Add linear fit with confidence interval
  labs(
    x = '',
    y = ""
  ) +
  theme_minimal() +
  theme(legend.title = element_blank()) +
  theme(legend.position = "none") +  # Remove legend
  theme(panel.background = element_blank()) + 
  ggpubr::stat_compare_means(comparisons = list(
    c('Control', 'ER+')
   #  c('Control', 'ER-'),
 #   c('ER+', 'ER-')
    ),label = 'p.format', label.y = c(1.6),step.increase = 0.13, vjust = 0,textsize=1)  +
  scale_fill_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control)) +
   scale_y_continuous(expand = expansion(mult = c(0, 0.1)))

k3

# Save coordinates
coord_save_excel(k3, coordPath) 
```

## l) WID-cervical: Association with PR status 


```{r}


l3 <- cervical_val |> 
  dplyr::mutate(PRstat = ifelse(type == 'Control', 'Control', ifelse(PR == 'Pos', 'PR+', 'PR-')),
                PRstat = factor(PRstat, levels = c("Control", "PR+", "PR-")))  |> 
  dplyr::filter(PRstat != 'Unknown') |> 
  ggplot(aes(x = PRstat,
             y = WID_cervical_BC_adj)) +
  geom_boxplot(aes(fill = type),
               alpha = 0.2,outlier.shape = NA) +
  ggbeeswarm::geom_quasirandom(aes(colour = type),
                               alpha = 0.5, size = 0.5) +
  scale_color_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control)) +  # Add linear fit with confidence interval
  labs(
    x = '',
    y = ""
  ) +
  theme_minimal() +
  theme(legend.title = element_blank()) +
  theme(legend.position = "none") +  # Remove legend
  theme(panel.background = element_blank()) + 
  ggpubr::stat_compare_means(comparisons = list(
    c('Control', 'PR+')
   #  c('Control', 'PR-'),

  #  c('PR+', 'PR-')
    ),label = 'p.format', label.y = c(1.6),step.increase = 0.13, vjust = 0,textsize=1)  +
  scale_fill_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control)) +
   scale_y_continuous(expand = expansion(mult = c(0, 0.1)))

l3

# Save coordinates
coord_save_excel(l3, coordPath) 
```

## m) WID-cervical: Association with HER2 status 


```{r}

m3 <- cervical_val |> 
  dplyr::mutate(HERstat = ifelse(type == 'Control', 'Control',
                                 ifelse(HER2 == 'Pos', 'HER2+', "HER2-")),
                HERstat = factor(HERstat, levels = c("Control", "HER2-", "HER2+")))  |> 
  dplyr::filter(HERstat != 'Unknown') |> 
  ggplot(aes(x = HERstat,
             y = WID_cervical_BC_adj)) +
  geom_boxplot(aes(fill = type),
               alpha = 0.2,outlier.shape = NA) +
  ggbeeswarm::geom_quasirandom(aes(colour = type),
                               alpha = 0.5, size = 0.5) +
  scale_color_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control)) +  # Add linear fit with confidence interval
  labs(
    x = '',
    y = "WID-cervical-BC index\nadjusted for age and ic"
  ) +
  theme_minimal() +
  theme(legend.title = element_blank()) +
  theme(legend.position = "none", axis.text.x = element_text(angle = 45, hjust = 1)) +  # Remove legend
  theme(panel.background = element_blank()) + 
  ggpubr::stat_compare_means(comparisons = list(
    c('Control', 'HER2-'),
     c('Control', 'HER2+')

  #  c('HER2+', 'HER2-')
    ),label = 'p.format', label.y = c(1.6),step.increase = 0.13, vjust = 0,textsize=1)  +
  scale_fill_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control)) +
   scale_y_continuous(expand = expansion(mult = c(0, 0.1)))

m3

# Save coordinates
coord_save_excel(m3, coordPath) 
```

## n) WID-cervical: PRS with WID-BC-cervical

```{r}


n3 <- ggplot(cervical_val, aes(x = polygenic_risk_score_bc, y = WID_cervical_BC_adj, color = type, fill = factor(type))) +
  geom_point(alpha=0.4,size=1) +  # Plot points
  geom_smooth(method = "lm", se = TRUE, aes(group = type)) +
  scale_color_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control)) +  # Add linear fit with confidence interval
  labs(
    y = "",
    x = "Polygenic Risk \n Score") + theme_minimal() +
  theme(legend.title = element_blank()) +
  theme(legend.position = "none") +  # Remove legend
  theme(panel.background = element_blank()) +  
  stat_cor(method = "pearson", show.legend = FALSE, label.x.npc='left', label.y.npc='bottom', size = 2.8) + 
  scale_fill_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control)) 

n3


  # Save coordinates
coord_save_excel(n3, coordPath) 

```

## o) WID-cervical: association with BRCAmut status

```{r}
brca_val$type <- factor(brca_val$type, levels = c('Control', 'BRCA1','BRCA2')) # Here we change the order of the levels

brca_cervical <- droplevels(brca_val[brca_val$sampletype=='cervical',])

# 1: Boxplot of WID-BC-cervical
o3 <- brca_cervical |> 
  ggplot(aes(x = factor(type), y = WID_cervical_BC_adj))  +
  geom_boxplot(aes(fill = type),
               alpha = 0.3,outlier.shape = NA) +
  ggbeeswarm::geom_quasirandom(aes(colour = type),
                               alpha = 0.5, size = 0.5) +  # Add beeswarm plot
  labs(x = "", y = "") +
  theme_minimal() +
  theme(legend.position = "none", axis.text.x = element_text(angle = 45, hjust = 1)) +
  # ggpubr::stat_compare_means(comparisons = list(
  #   c("Control", "BRCA1"),
  #   c("Control", "BRCA2"),
  #   c("BRCA1", "BRCA2")
  # ), label = 'p.signif') +
  guides(fill = guide_legend(title.position = "top", title.hjust = 0.5)) + 
  scale_fill_manual(values = c("Control" = colour.Control, "BRCA1" = pal.jama[5], "BRCA2" = pal.jama[5]),
                    aesthetics = c('colour', 'fill')) 


o3

  # Save coordinates
coord_save_excel(o3, coordPath) 

```

## p) WID-cervical: AUC of WID-BC-cervical in BRCA1/BRCA2 mutation carriers


```{r}
filtered_BRCA2 <- brca_cervical |>
  filter(type == 'Control' | type == 'BRCA2')
filtered_BRCA1 <- brca_cervical |>
  filter(type == 'Control' | type == 'BRCA1')

# Here we change the order of the levels, such that AUC calculation goes right.
filtered_BRCA2$type <- factor(filtered_BRCA2$type, levels = c('Control', 'BRCA2')) 
filtered_BRCA1$type <- factor(filtered_BRCA1$type, levels = c('Control', 'BRCA1')) 


p3 <- plot_double_roc(factor(filtered_BRCA1$type), filtered_BRCA1$WID_cervical_BC_adj, factor(filtered_BRCA2$type), filtered_BRCA2$WID_cervical_BC_adj, col1 = colour.blue, col2 = colour.green, title1 = "BRCA1", title2 = "BRCA2", style = "default",
                            direction1 = "<",direction2 = "<", generalTitle="",textSize = 2.1) 

p3

  # Save coordinates
coord_save_excel(p3, coordPath) 
```

## q) WID-blood: association with BRCAmut status

```{r}
brca_val$type <- factor(brca_val$type, levels = c('Control', 'BRCA1','BRCA2')) # Here we change the order of the levels

brca_blood <- droplevels(brca_val[brca_val$sampletype=='blood',])

# 1: Boxplot of WID-BC-cervical
q3 <- brca_blood |> 
  ggplot(aes(x = factor(type), y = WID_blood_BC_adj))  +
  geom_boxplot(aes(fill = type),
               alpha = 0.3,outlier.shape = NA) +
  ggbeeswarm::geom_quasirandom(aes(colour = type),
                               alpha = 0.5, size = 0.5) +  # Add beeswarm plot
  labs(x = "", y = "WID-blood-BC index\nadjusted for neutrophils") +
  theme_minimal() +
  theme(legend.position = "none", axis.text.x = element_text(angle = 45, hjust = 1)) +
  # ggpubr::stat_compare_means(comparisons = list(
  #   c("Control", "BRCA1"),
  #   c("Control", "BRCA2"),
  #   c("BRCA1", "BRCA2")
  # ), label = 'p.signif') +
  guides(fill = guide_legend(title.position = "top", title.hjust = 0.5)) + 
  scale_fill_manual(values = c("Control" = colour.Control, "BRCA1" = pal.jama[5], "BRCA2" = pal.jama[5]),
                    aesthetics = c('colour', 'fill'))


q3

  # Save coordinates
coord_save_excel(q3, coordPath) 

```

## r) WID-blood: AUC of WID-BC-blood in BRCA1/BRCA2 mutation carriers


```{r}
filtered_BRCA2 <- brca_blood |>
  filter(type == 'Control' | type == 'BRCA2')
filtered_BRCA1 <- brca_blood |>
  filter(type == 'Control' | type == 'BRCA1')

# Here we change the order of the levels, such that AUC calculation goes right.
filtered_BRCA2$type <- factor(filtered_BRCA2$type, levels = c('Control', 'BRCA2')) 
filtered_BRCA1$type <- factor(filtered_BRCA1$type, levels = c('Control', 'BRCA1')) 


r3 <- plot_double_roc(factor(filtered_BRCA1$type), filtered_BRCA1$WID_blood_BC_adj, factor(filtered_BRCA2$type), filtered_BRCA2$WID_blood_BC_adj, col1 = colour.blue, col2 = colour.green, title1 = "BRCA1", title2 = "BRCA2", style = "default",
                            direction1 = "<",direction2 = "<", generalTitle="",textSize = 2.1)

r3

  # Save coordinates
coord_save_excel(r3, coordPath) 
```





## Compile and save

```{r, message=FALSE, echo=FALSE}

# Will need to be amended with updated plot
layout <- "
ABCD
EFGH
IJKL
MNOP
QRS#
"


pdf_width <- 8.3  # Width in inches (a4)
pdf_height <- 11.7   # Height in inches (a4: 11.7)


compiled <- a3 + b3 + c3 + d3 + e3 + f3 + g3 + h3+ i3 + j3 + k3+ l3 + m3 + n3 + o3 + p3 + q3 + r3 + guide_area()
  
compiled <-  compiled + plot_annotation(tag_levels = 'a') + 
  plot_layout(design = layout, guides = 'collect')

ggsave(file.path(outPath,"F3.pdf"), width = pdf_width, height = pdf_height, compiled) # fix labels in illustrator
# ggsave(file.path(jpgPath,"F3.jpg"), width = pdf_width, height = pdf_height, compiled)
```



# Figure 4

## a) Boxplot tissue at risk

```{r}
library(dplyr)
library(ggh4x)

# Rename BRCA1/BRCA2 
tissue_at_risk_val <- tissue_at_risk_val |>
  mutate(type = ifelse(type %in% c("Risk_reduction_BRCA1", "Risk_reduction_BRCA2"), "Risk_reduction_BRCA1_BRCA2", type))


# Long format
tissue_at_risk_long <- tidyr::pivot_longer(tissue_at_risk_val,
                                    cols = c(WID_buccal_BC_adj, WID_cervical_BC_adj,WID_blood_BC_adj),
                                    names_to = "index",
                                    values_to = "score"
)


# id for pairing
tissue_at_risk_long <- tissue_at_risk_long |> 
  mutate(pairs = ifelse(!grepl("adjacent|negative", type), NA, id))

# change order of indices for plotting
tissue_at_risk_long$index <- factor(tissue_at_risk_long$index, levels=c("WID_buccal_BC_adj","WID_cervical_BC_adj","WID_blood_BC_adj"))

# change order of type for plotting
tissue_at_risk_long$type <- factor(tissue_at_risk_long$type, levels = c("Normal", "Risk_reduction_BRCA1_BRCA2", "Normal_adjacent", "Triple_negative_tumour"))

# Only colour strips in x-direction
strip <- strip_themed(background_x = elem_list_rect(fill = c(colour.green, colour.blue, colour.red),
                                                    color = NA),
                      text_x = element_text(colour = 'white',
                                            face = 'bold'))

cols <- ggsci::pal_npg()(4)

# Facet wrap by index
a4 <- ggplot(tissue_at_risk_long,
             aes(x = type,
                 y = score)) +
  
  geom_boxplot(aes(fill = type),
               cex = 0.5, fatten = 2,
               outlier.shape = NA,
               alpha = 0.4) +
  
  ggbeeswarm::geom_beeswarm(aes(colour = type),
                            alpha = 0.8, size = 1) +
  
  labs(
    x = "",
    y = "Index"
  )   + 
  
  # geom_line(aes(group=pairs), size=0.5, color='gray', alpha=0.6) +
  theme_minimal() +
  
  scale_fill_manual(values = rev(cols),
                    aesthetics = c("colour", "fill"),
                 labels = c("Normal tissue",
                 "Normal from\nBRCA1/BRCA2\nmutation carrier",
                 "Normal-adjacent\ntissue", "Triple negative\nbreast cancer\ntissue"),
                 name = '') +
  
  theme(legend.position = "right", 
        axis.text.x = element_blank()) +
  
  facet_wrap2(~index, labeller = labeller(index = c("WID_buccal_BC_adj" = "WID-buccal-BC",
                                                    "WID_cervical_BC_adj" = "WID-cervical-BC",
                                                    "WID_blood_BC_adj" = "WID-blood-BC")),
             scales = 'free',
             strip = strip) +
  
  ggpubr::stat_compare_means(comparisons = list(
   c("Normal", "Risk_reduction_BRCA1_BRCA2"), 
  #  c("Normal", "Normal_adjacent"), # insignificant in all
    c("Normal", "Triple_negative_tumour")),
                               label = 'p.format',
  step.increase = 0.13, vjust = 0,
  size=3.2, hide.ns = T) +
  labs(y = '',
       title = 'Tissue at risk set (EPIC array)') +
   scale_y_continuous(expand = expansion(mult = c(0, 0.1)))

print(a4)

# Save coordinates
coord_save_excel(a4, coordPath) 
```

## b) AUC tissue at risk normal-adjacent

```{r}
## Normal-adjacent versus triple negative
tmp <- tissue_at_risk_val |> 
  dplyr::filter(type %in% c("Normal_adjacent","Triple_negative_tumour")) |> 
  droplevels()

b4 <- plot_triple_roc(factor(tmp$type), tmp$WID_buccal_BC_adj, factor(tmp$type), tmp$WID_cervical_BC_adj, factor(tmp$type), tmp$WID_blood_BC_adj, col1 = colour.green, col2 = colour.blue, col3 = colour.red, title1 = "WID-buccal-BC", title2 = "WID-cervical-BC", title3 = "WID-blood-BC", style = "default",
                            direction1 = "<",direction2 = "<", generalTitle="", textCol1 = "white",textSize=2.1,y3=0.35,y2=0.1)

b4

  # Save coordinates
coord_save_excel(b4, coordPath) 

```

## c) Boxplot TCGA

```{r}
# Long format
tcga_long <- tidyr::pivot_longer(tcga_val,
  cols = c(WID_buccal_BC_adj, WID_cervical_BC_adj, WID_blood_BC_adj),
  names_to = "index",
  values_to = "score"
)



# Keep paired only
tmp <- tcga_long |>
  dplyr::group_by(barcode3) |>
  dplyr::filter(all(c("Control", "Cancer") %in% type)) 



# Reorder the 'type' variable to have 'Control' first
tcga_long$type <- factor(tcga_long$type, levels = c('Control', 'Cancer'))
tcga_long$index <- factor(tcga_long$index, levels = c('WID_buccal_BC_adj', 'WID_cervical_BC_adj','WID_blood_BC_adj'))

# facet wrap by index
c4 <- tcga_long |>
  dplyr::filter(barcode3 %in% tmp$barcode3) |>
  
  ggplot(aes(x = type,
             y = score)) +
  
  geom_boxplot(aes(fill = type),
               cex = 0.5, fatten = 2,
               outlier.shape = NA,
               alpha = 0.4) +
  geom_line(aes(group = barcode3),
            size = 0.2, colour = 'grey80') +
  
  ggbeeswarm::geom_beeswarm(aes(colour = type),
                            alpha = 0.8, size = 1) +
  
  labs(
    x = "",
    y = "Index"
  )   + 
  
  # geom_line(aes(group=pairs), size=0.5, color='gray', alpha=0.6) +
  theme_minimal() +
  
  scale_fill_manual(values = rev(cols)[c(3, 4)],
                    aesthetics = c("colour", "fill"),
                 labels = c("Matched\nnormal tissue", 'Breast cancer\ntissue'),
                 name = '') +
  
  theme(legend.position = "right", 
        axis.text.x = element_blank()) +
  
  facet_wrap2(~index, labeller = labeller(index = c("WID_buccal_BC_adj" = "WID-buccal-BC",
                                                    "WID_cervical_BC_adj" = "WID-cervical-BC",
                                                    "WID_blood_BC_adj" = "WID-blood-BC")),
             scales = 'free',
             strip = strip) +
  
  ggpubr::stat_compare_means(comparisons = list(c("Control", "Cancer")),
                               label = 'p.format',
  step.increase = 0.13, vjust = 0,
  size=3.2, hide.ns = T) +
  labs(y = '',
       title = 'TCGA Set (450K array)') +
   scale_y_continuous(expand = expansion(mult = c(0, 0.1)))

print(c4)

# Save coordinates
coord_save_excel(c4, coordPath) 
```


## d) AUC TCGA

```{r}
# Reorder the 'type' variable to have 'Control' first
tcga_val$type <- factor(tcga_val$type, levels = c('Control', 'Cancer'))

d4 <- plot_triple_roc(factor(tcga_val$type), tcga_val$WID_buccal_BC_adj, factor(tcga_val$type), tcga_val$WID_blood_BC_adj, factor(tcga_val$type), tcga_val$WID_cervical_BC_adj, col1 = colour.green, col2 = colour.red, col3 = colour.blue, title1 = "WID-buccal-BC", title2 = "WID-blood-BC", title3 = "WID-cervical-BC", style = "default",
                            direction1 = "<",direction2 = "<", generalTitle="", textCol1 = "white",textSize=2.1)

d4

# Save coordinates
coord_save_excel(d4, coordPath) 


```

## e) Boxplot GSE225845

```{r}
# Long format
GSE225845_long <- tidyr::pivot_longer(GSE225845_val,
                                    cols = c(WID_buccal_BC_adj, WID_cervical_BC_adj,WID_blood_BC_adj),
                                    names_to = "index",
                                    values_to = "score"
)

GSE225845_long$type <- factor(GSE225845_long$type, levels=c("normal","adj_norm","tumor"))

# change order of indices for plotting
GSE225845_long$index <- factor(GSE225845_long$index, levels=c("WID_buccal_BC_adj","WID_cervical_BC_adj","WID_blood_BC_adj"))

# Facet wrap by index
e4 <- GSE225845_long |> 
  
  ggplot(aes(x = type,
             y = score)) +
  
  geom_boxplot(aes(fill = type),
               cex = 0.5, fatten = 2,
               outlier.shape = NA,
               alpha = 0.4) +
  
  ggbeeswarm::geom_beeswarm(aes(colour = type),
                            alpha = 0.6, size = 1) +
  
  labs(
    x = "",
    y = "Index"
  )   + 
  theme_minimal() +
  
  scale_fill_manual(values = rev(cols)[c(1, 3, 4)],
                    aesthetics = c("colour", "fill"),
                 labels = c("Normal", "Normal\nadjacent", "Tumor"),
                 name = '') +
  
  theme(legend.position = "right", 
        axis.text.x = element_blank()) +
  
  facet_wrap2(~index, labeller = labeller(index = c("WID_buccal_BC_adj" = "WID-buccal-BC",
                                                    "WID_cervical_BC_adj" = "WID-cervical-BC",
                                                    "WID_blood_BC_adj" = "WID-blood-BC")),
             scales = 'free',
             strip = strip) +
  
  ggpubr::stat_compare_means(comparisons = list(
    c("normal", "tumor"),
    c("normal", "adj_norm"),
    c("adj_norm", "tumor")),
                               label = 'p.format',
  step.increase = 0.13, vjust = 0,
  size=3.2, hide.ns = T) +
  labs(y = '',
       title = 'GSE225845 (EPIC array)') +
   scale_y_continuous(expand = expansion(mult = c(0, 0.1)))
  
  

print(e4)

  # Save coordinates
coord_save_excel(e4, coordPath) 
```

## f) AUC GSE225845 normal-adjacent

```{r}

## Normal-adjacent versus triple negative
tmp <- GSE225845_val |> 
  dplyr::filter(type %in% c("normal","tumor")) |> 
  droplevels()

f4 <- plot_triple_roc(factor(tmp$type), tmp$WID_buccal_BC_adj, factor(tmp$type), tmp$WID_cervical_BC_adj, factor(tmp$type), tmp$WID_blood_BC_adj, col1 = colour.green, col2 = colour.blue, col3 = colour.red, title1 = "WID-buccal-BC", title2 = "WID-cervical-BC", title3 = "WID-blood-BC", style = "default",
                            direction1 = "<",direction2 = "<", generalTitle="", textCol1 = "white",textSize=2.1,y3=0.35,y2=0.1)

f4

  # Save coordinates
coord_save_excel(f4, coordPath) 
```


## Compile and save
```{r, message=FALSE, echo=FALSE}
pdf_width <- 10  # Width in inches (a4: 8.3)
pdf_height <- 9   # Height in inches (a4: 11.7)


layout <- "
AAAAAABB
CCCCCCDD
EEEEEEFF
"
compiled <- a4 + b4 + c4 + d4 + e4 + f4
  
compiled <-  compiled + plot_annotation(tag_levels = 'a') + 
  plot_layout(design = layout) &
    theme(plot.tag = element_text(face = 'bold'))

ggsave(file.path(outPath,"F4.pdf"), width = pdf_width, height = pdf_height, compiled)
# ggsave(file.path(jpgPath,"F4.jpg"), width = pdf_width, height = pdf_height, compiled)
```

#=========SUPPLEMENTARY TABLES========

# ST1: Epigenome-wide study buccal samples
.csv file

```{r}
library(IlluminaHumanMethylationEPICanno.ilm10b4.hg19)

buccal <- readRDS(here("1-ewas", "1-buccal", "delta-beta.Rds")) |> 
  as.data.frame() |> 
  dplyr::mutate(padj = p.adjust(type, method = 'BH')) |> 
  tibble::rownames_to_column('cg')

# Get annotation
man.epic <- as.data.frame(getAnnotation(IlluminaHumanMethylationEPICanno.ilm10b4.hg19))

# Reformat the multiple names to first value
man.epic$UCSC_RefGene_Name <- stringr::str_split(man.epic$UCSC_RefGene_Name, ";", simplify = T)[,1]

# Find CpGs
anno <- match(buccal$cg, man.epic$Name)

# Extract info delta-beta
tmp <- man.epic |>
  select(Name, chr, pos, UCSC_RefGene_Name, UCSC_RefGene_Group, Relation_to_Island)

# Subset the selected columns based on the 'anno' indices
result <- cbind(buccal,tmp[anno, ])

# rename column
result$pvalue <- result$type

# Select relevant columns for supplement
result <- result |>
  select(cg, chr, pos, dir_type, db_epithelial, db_immune, pvalue, padj, UCSC_RefGene_Name, UCSC_RefGene_Group, Relation_to_Island)

# Save to .csv
write.csv(result, file.path(supPath, "ST1.csv"), row.names=FALSE)

# # Save a separate .csv file containing only those with padj<0.05
# write.csv(result[result$padj<0.05,], file.path(supPath, "ST3_cropped.csv"), row.names=FALSE)


```

# ST2: Epigenome-wide study cervical samples

.csv file

```{r}
cervical <- readRDS(here("1-ewas", "2-cervical", "delta-beta.Rds")) |> 
  as.data.frame() |> 
  dplyr::mutate(padj = p.adjust(type, method = 'BH')) |> 
  tibble::rownames_to_column('cg')


# Get annotation
man.epic <- as.data.frame(getAnnotation(IlluminaHumanMethylationEPICanno.ilm10b4.hg19))

# Reformat the multiple names to first value
man.epic$UCSC_RefGene_Name <- stringr::str_split(man.epic$UCSC_RefGene_Name, ";", simplify = T)[,1]

# Find CpGs
anno <- match(cervical$cg, man.epic$Name)

# Extract info delta-beta
tmp <- man.epic |>
  select(Name, chr, pos, UCSC_RefGene_Name, UCSC_RefGene_Group, Relation_to_Island)

# Subset the selected columns based on the 'anno' indices
result <- cbind(cervical,tmp[anno, ])

# rename column
result$pvalue <- result$type


# Select relevant columns for supplement
result <- result |>
  select(cg, chr, pos, dir_type, db_epithelial, db_immune, pvalue, padj, UCSC_RefGene_Name, UCSC_RefGene_Group, Relation_to_Island)

# Save to .csv
write.csv(result, file.path(supPath, "ST2.csv"), row.names=FALSE)

# Save a separate .csv file containing only those with padj<0.05
# write.csv(result[result$padj<0.05,], file.path(supPath, "ST4_cropped.csv"), row.names=FALSE)
```

# ST3: Epigenome-wide study cervical samples
.csv file

```{r}
blood <- readRDS(here("1-ewas", "3-blood", "delta-beta.Rds")) |> 
  as.data.frame() |> 
  dplyr::mutate(padj = p.adjust(type, method = 'BH')) |> 
  tibble::rownames_to_column('cg')


# Get annotation
man.epic <- as.data.frame(getAnnotation(IlluminaHumanMethylationEPICanno.ilm10b4.hg19))

# Reformat the multiple names to first value
man.epic$UCSC_RefGene_Name <- stringr::str_split(man.epic$UCSC_RefGene_Name, ";", simplify = T)[,1]

# Find CpGs
anno <- match(blood$cg, man.epic$Name)

# Extract info delta-beta
tmp <- man.epic |>
  select(Name, chr, pos, UCSC_RefGene_Name, UCSC_RefGene_Group, Relation_to_Island)

# Subset the selected columns based on the 'anno' indices
result <- cbind(blood,tmp[anno, ])

# rename column
result$pvalue <- result$type


# Select relevant columns for supplement
result <- result |>
  select(cg, chr, pos, dir_type, db_epithelial, db_immune, pvalue, padj, UCSC_RefGene_Name, UCSC_RefGene_Group, Relation_to_Island)

# Save to .csv
write.csv(result, file.path(supPath, "ST3.csv"), row.names=FALSE)
```

# ST4: Overlapping genomic regions, buccal and cervical

```{r}
# Load data
load(here("1-ewas", "1-buccal", "2-output", "ranges.Rdata")) 
buccal <- ranges

load(here("1-ewas", "2-cervical", "2-output", "ranges.Rdata")) 
cervical <- ranges

buccal <- sortSeqlevels(buccal)
buccal <- sort(buccal)
# buccal <- buccal[buccal$Stouffer<0.05,]

cervical <- sortSeqlevels(cervical)
cervical <- sort(cervical)
# cervical <- cervical[cervical$Stouffer,]

overlaps1 <- subsetByOverlaps(buccal, cervical)
overlaps2 <- subsetByOverlaps(cervical, buccal)

# Supplementary table for overlaps
overlaps <- as.data.frame(overlaps1) |> 
  dplyr::select(seqnames, start, end, width, strand, no.cpgs, overlapping.genes, meandiff) |> 
  dplyr::rename(meandiff_buccal = meandiff)

overlaps$meandiff_cervical <- overlaps2$meandiff

# Overlaps object can be exported as supplementary table

# Save to .csv
write.csv(overlaps, file.path(supPath, "ST6.csv"), row.names=FALSE)
```



# ST5: BRCA table

```{r}
theme_gtsummary_journal(journal = "jama")
theme_gtsummary_compact()


tmp1 <- data |> 
  # Mutate
  dplyr::filter(set %in% c("validation_genetic_risk")) |> 
  
  # Mutate for formatting
  dplyr::mutate(type = case_when(type == "BC case" ~ "BC",
                                 type == "Control" ~ "CO",
                                 TRUE ~ type),
                sampletype = case_when(sampletype == "buccal" ~ "Buccal",
                                    sampletype == "cervical" ~ "Cervical",
                                    sampletype == "blood" ~ "Blood"),
                sampletype = factor(sampletype, levels = c("Buccal",
                                                     "Cervical",
                                                     "Blood")))

# Columns select
tmp1 <- tmp1 |> select(sampletype, age, type, BMI, smoking)

# Replace empty columns with NA
tmp1 <- tmp1 |> mutate_if(is.character, ~na_if(., ''))

tmp1$smoking<- as.character(tmp1$smoking)

tbl <- tmp1 |>
  tbl_strata(strata = sampletype,
             .tbl_fun= 
               ~ .x |>
               tbl_summary(by = 'type',
                           type = list(age ~ "continuous",BMI ~ "continuous"),
                           statistic = list(age ~ "{mean} ({sd})",BMI ~ "{mean} ({sd})"),
                           missing = "no",
                           label = list(age = "Age at sample taken",
                                        type = "Type",
                                        BMI = "Body Mass Index",
                                        smoking = "Current smoker"
                                        )) |>
               bold_labels() |>
               modify_header(update = all_stat_cols() ~"**{level}**<br>n = {n}")) 


t1 <- tbl |>
  as_gt()  |>
  text_transform(locations = cells_body(),
                 fn = function(x){
                   paste0(ifelse(x == "0 (0)", "", ifelse(x == "NA (NA – NA)", "", x)))
                 }
  ) |>
  text_transform(locations = cells_body(),
                 fn = function(x){
                   paste0(ifelse(x == "0.0000 (0.0000–0.0000)", "0 (0–0)", x))
                 }
  ) |>
  text_transform(locations = cells_body(),
                 fn = function(x){
                   paste0(ifelse(x == "0 (NA)", "", x))
                 }
  ) |>
  text_transform(locations = cells_body(),
                 fn = function(x){
                   paste0(ifelse(x == "NA (NA)", "", x))
                 }
  ) |>
  tab_options(
    table.width = px(650),
    table.font.size = 12,
    table.font.names = "Arial",
  ) |> 
  opt_stylize(style = 4)

t1 |> gtsave(file.path(outPath,"ST1.html"))

# to pdf
pagedown::chrome_print(file.path(outPath,"ST1.html"), output = file.path(outPath,"ST1.pdf"))
```

# ST6: Breast datasets

```{r}
theme_gtsummary_journal(journal = "jama")
theme_gtsummary_compact()

load(here("0-data", "GSE22845","pheno_trimmed.Rdata"))# Load pheno file from additional breast dataset
load(here("0-data", "dataframes-plotting","tcga_validation.Rdata")) # load pheno file from TCGA

pheno_trimmed$set <- "validation_breast"
pheno_trimmed$subset <- "GSE225845"
pheno_trimmed$stage_T <- pheno_trimmed$stage  
pheno_trimmed$ER <- pheno_trimmed$er

tcga_val <- tcga_val |> dplyr::mutate(type = case_when(type == "Control" ~ "Normal breast tissue adjacent to a cancer", type == "Cancer" ~ "Breast cancer tissue"))
tcga_val$set <- "validation_breast"
tcga_val$subset <- "TCGA"

# Merge data frames by matching column names
data1 <- merge(data, pheno_trimmed, by = intersect(names(data), names(pheno_trimmed)), all.x = TRUE, all.y=TRUE)
data2 <- merge(data1, tcga_val, by = intersect(names(data), names(tcga_val)), all.x = TRUE, all.y=TRUE)

tmp1 <- data2 |> 
  # Mutate
  dplyr::filter(set %in% c("validation_breast")) |> 
  
  # Mutate for formatting
  dplyr::mutate(type = case_when(type == "Normal_adjacent" ~ "Normal breast tissue adjacent to a cancer",
                                 type == "adj_norm" ~ "Normal breast tissue adjacent to a cancer",type == "Normal breast tissue adjacent to a cancer" ~ "Normal breast tissue adjacent to a cancer",
                                 type == "Triple_negative_tumour" ~ "Breast cancer tissue",
                                 type == "Cancer" ~ "Breast cancer tissue",type == "Breast cancer tissue" ~ "Breast cancer tissue",
                                 type == "tumor" ~ "Breast cancer tissue",
                                 type == "Normal" ~ "Normal breast tissue from healthy control",
                                 type == "Control" ~ "Normal breast tissue from healthy control",
                                 type == "normal" ~ "Normal breast tissue from healthy control",
                                 type == "Risk_reduction_BRCA1" ~ "Normal tissue from BRCA1 mutation carrier",
                                 type == "Risk_reduction_BRCA2" ~ "Normal tissue from BRCA2 mutation carrier",
                                 TRUE ~ type),
                subset = case_when(subset == "breast_tissue_epic" ~ "Tissue at risk<br>(EPIC array)",
                                   subset == "TCGA" ~ "TCGA<br>(450K array)",
                                   subset == "GSE225845" ~ "GSE225845<br>(EPIC array)"))

# Stage T cleaning
tmp1 <- tmp1 |>
  mutate(stage_T = case_when(
    is.na(stage_T)~ NA,
    grepl("^T[0-9]$", stage_T) ~ stage_T,  # Entries already in T[number] format
    grepl("^T[0-9]a$", stage_T) ~ gsub("a$", "", stage_T),  # Remove 'a' from T[number]a
    grepl("^T[0-9]b$", stage_T) ~ gsub("b$", "", stage_T),  # Remove 'b' from T[number]b
    grepl("^T[0-9]c$", stage_T) ~ gsub("c$", "", stage_T),  # Remove 'c' from T[number]c
    grepl("^T[0-9]d$", stage_T) ~ gsub("d$", "", stage_T),  # Remove 'c' from T[number]d
    stage_T == "I" ~ "T1",  
    stage_T == "II" ~ "T2",  
    stage_T == "III" ~ "T3",  
    stage_T == "IV" ~ "T4",  
    stage_T == "TIS" ~ "Tis",  
    stage_T == "TX" ~ "Unknown",  # Rename TX to Unknown
    TRUE ~ as.character(stage_T)  # Keep other entries as is
  ))

# Stage N cleaning
tmp1 <- tmp1 |>
  mutate(stage_N = case_when(stage_N == "N0 (i-)" ~ "N0",  
         stage_N == "N0 (i+)" ~ "N0",
         stage_N == "N0 (mol+)" ~ "N0",
         stage_N == "N1a" ~ "N1",
         stage_N == "N1b" ~ "N1",
         stage_N == "N1c" ~ "N1",
         stage_N == "N1mi" ~ "N1",
         stage_N == "N2a" ~ "N2",
         stage_N == "N3a" ~ "N3",
         stage_N == "N3b" ~ "N3",
         stage_N == "NX" ~ "Unknown"
  ))


# Convert to 'Unknown' for BC 
# Stage T
tmp1$stage_T[is.na(tmp1$stage_T) & tmp1$type=="Breast cancer tissue" & tmp1$subset!="Tissue at risk<br>(EPIC array)"] <- "Unknown"

# Stage N
tmp1$stage_N[is.na(tmp1$stage_N)  & tmp1$type=="Breast cancer tissue"& tmp1$subset!="Tissue at risk<br>(EPIC array)"] <- "Unknown"


# grade
tmp1$grade[is.na(tmp1$grade)  & tmp1$type=="Breast cancer tissue"& tmp1$subset!="Tissue at risk<br>(EPIC array)"] <- "Unknown"

# ER
tmp1$ER[is.na(tmp1$ER)  &tmp1$type=="Breast cancer tissue"& tmp1$subset!="Tissue at risk<br>(EPIC array)"] <- "Unknown"

# PR
tmp1$PR[is.na(tmp1$PR)  &tmp1$type=="Breast cancer tissue" & tmp1$subset!="Tissue at risk<br>(EPIC array)"] <- "Unknown"
#

# Columns select
tmp1 <- tmp1 |> select(subset, age, type, stage_T, stage_N, grade, ER)


# Columns to convert to factors
columns_to_factor <- c("subset","type", "stage_T",   "stage_N",   "grade" ,    "ER")

# Convert selected columns to factors
tmp1[, columns_to_factor] <- lapply(tmp1[, columns_to_factor], factor)


# Replace empty columns with NA
tmp1 <- tmp1 |> mutate_if(is.character, ~na_if(., ''))


tbl <- tmp1 |>
  tbl_strata(strata = subset,
             .tbl_fun= 
               ~ .x |>
               tbl_summary(type = list("age" ~ "continuous"),
                           statistic = list("age" ~ "{mean} ({sd})"),
                           missing = "no",
                           label = list("age" = "Age at sample taken",
                                        "type" = "Type",
                                        "stage_T" = "Stage (T)",
                                        "stage_N" = "Stage (N)",
                                        "grade" = "Grade")) |>
               bold_labels() |>
               modify_header(all_stat_cols() ~"**{level}**<br>n = {n}")) 


t1 <- tbl |>
  as_gt()  |>
  text_transform(locations = cells_body(),
                 fn = function(x){
                   paste0(ifelse(x == "0 (0)", "-", ifelse(x == "NA (NA – NA)", "-", x)))
                 }
  ) |>
  text_transform(locations = cells_body(),
                 fn = function(x){
                   paste0(ifelse(x == "0.0000 (0.0000–0.0000)", "0 (0–0)", x))
                 }
  ) |>
  text_transform(locations = cells_body(),
                 fn = function(x){
                   paste0(ifelse(x == "0 (NA)", "-", x))
                 }
  ) |>
  text_transform(locations = cells_body(),
                 fn = function(x){
                   paste0(ifelse(x == "NA (NA)", "-", x))
                 }
  ) |>
  tab_options(
    table.width = px(650),
    table.font.size = 12,
    table.font.names = "Arial",
  ) |> 
  opt_stylize(style = 4)

t1 |> gtsave(file.path(outPath,"ST2.html"))

# to pdf
pagedown::chrome_print(file.path(outPath,"ST2.html"), output = file.path(outPath,"ST2.pdf"))
```

#=========EXTENDED DATA FIGURES ========

# Extended Data 1: IC and cell types in the discovery and external validation sets



## a) Buccal: differences in IC

```{r}
subset_data <- subset(data, set %in% c("discovery_buccal", "validation_buccal"))
subset_data$type <- factor(subset_data$type, levels=c("Control","BC case"))

ED1a <- ggplot(data=subset_data, aes(x = type,
             y = ic,
            fill = type)) +
  geom_boxplot(aes(fill = type),
               alpha = 0.2,outlier.shape = NA) +
  ggbeeswarm::geom_quasirandom(aes(colour = type),
                               alpha = 0.5, size=0.5) +
  scale_color_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control)) +
  theme_minimal() +
  theme(legend.title = element_blank(),
        legend.position = '',
        axis.text.x = element_blank(),
        axis.title.y = element_markdown()) +
  scale_fill_manual(values = c(colour.Control, colour.Breast)) +
  labs(x = "", y = "Immune Cell Fraction", title= "Buccal") + 
  ggpubr::stat_compare_means(comparisons = list(c("BC case","Control")),
                             label = 'p.format', label.y = c(0.75), vjust = +0.1,size = 3.2) +
  facet_wrap(~set,labeller = as_labeller(c(`discovery_buccal` = "Discovery\nSet", `validation_buccal` = "Validation\nSet"))) + 
       theme(strip.text.x = element_text(size = 8))

ED1a

```

## b) Cervical: differences in IC


```{r}
subset_data <- subset(data, set %in% c("discovery_cervical", "validation_cervical"))
subset_data$type <- factor(subset_data$type, levels=c("Control","BC case"))

ED1b <- ggplot(data=subset_data, aes(x = type,
             y = ic,
            fill = type)) +
  geom_boxplot(aes(fill = type),
               alpha = 0.2, outlier.shape = NA) +
  ggbeeswarm::geom_quasirandom(aes(colour = type),
                               alpha = 0.5, size=0.5) +
  scale_color_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control)) +
  theme_minimal() +
  theme(legend.title = element_blank(),
        legend.position = 'top',
        legend.direction = "vertical",
        axis.text.x = element_blank(),
        axis.title.y = element_markdown()) +
  scale_fill_manual(values = c(colour.Control, colour.Breast)) +
  labs(x = "", y = "Immune Cell Fraction", title= "Cervical") + 
  ggpubr::stat_compare_means(comparisons = list(c("BC case","Control")),
                             label = 'p.format', label.y = c(0.9), vjust = +0.1,size = 3.2) +
  facet_wrap(~set,labeller = as_labeller(c(`discovery_cervical` = "Discovery\nSet", `validation_cervical` = "Validation\nSet"))) + 
       theme(strip.text.x = element_text(size = 8))

ED1b

```

## c) Blood: differences in Neutro

```{r}
subset_data <- subset(data, set %in% c("discovery_blood"))
subset_data2 <- blood_val
subset_data <- plyr::rbind.fill(subset_data, blood_val)
subset_data$type <- factor(subset_data$type, levels=c("Control","BC case"))

ED1c <- ggplot(data=subset_data, aes(x = type,
             y = hepidish_Neutro,
            fill = type)) +
  geom_boxplot(aes(fill = type),
               alpha = 0.2, outlier.shape = NA) +
  ggbeeswarm::geom_quasirandom(aes(colour = type),
                               alpha = 0.5, size=0.5) +
  scale_color_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control)) +
  theme_minimal() +
  theme(legend.title = element_blank(),
        legend.position = '',
        axis.text.x = element_blank(),
        axis.title.y = element_markdown()) +
  scale_fill_manual(values = c(colour.Control, colour.Breast)) +
  labs(x = "", y = "Neutrophil Fraction", title= "Blood")  + 
  ggpubr::stat_compare_means(comparisons = list(c("BC case","Control")),
                             label = 'p.format', label.y = c(0.85), vjust = +0.1,size = 3.2) +
  facet_wrap(~set,labeller = as_labeller(c(`discovery_blood` = "Discovery\nSet", `validation_blood` = "Validation\nSet"))) + 
       theme(strip.text.x = element_text(size = 8))

ED1c
```

## d) Buccal: Cell type boxplot BC / control.

```{r}
subset_data <- subset(data, set %in% c("discovery_buccal", "validation_buccal"))
subset_data$type <- factor(subset_data$type, levels=c("Control","BC case"))

# Define the new facet labels
labs <- c("Basophils", "CD4T", "CD8T", "Eosinophils", "Epithelial", "Fibroblasts", "Monocytes", "Neutrophils", "NK cells")
names(labs) <- c("hepidish_B", "hepidish_CD4T", "hepidish_CD8T", "hepidish_Eosino", "hepidish_Epi", "hepidish_Fib", "hepidish_Mono", "hepidish_Neutro", "hepidish_NK")

labs_names <- c("Discovery Set", "Validation Set")
names(labs_names) <- c("discovery_buccal", "validation_buccal")

# Define the columns you want to gather
celltypes <- c("hepidish_B", "hepidish_CD4T", "hepidish_CD8T", "hepidish_Eosino", "hepidish_Epi", "hepidish_Fib", "hepidish_Mono", "hepidish_Neutro", "hepidish_NK")

# Reshape the dataframe to long format
subset_data_long <- subset_data |>
  gather(key = "cell_type", value = "Fraction", celltypes)


ED1d <- ggplot(subset_data_long, aes(x = type, y = Fraction, fill = type)) +
  geom_boxplot(outlier.shape = NA) +
  coord_flip() +
  facet_grid(cell_type~set, labeller = labeller(cell_type= labs, set=labs_names))  +
  labs(y = "Proportion", x = "Cell type", fill = "", title= "Buccal") + 
  ggpubr::stat_compare_means(ref.group = 'Control',
                             label = 'p.signif',
                             hide.ns = T, label.y = c(0.9), vjust = +0.7) +
  theme_minimal() +
  theme(legend.position = "none",  # Move the legend to the bottom
        legend.direction = "horizontal",
        panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    axis.text.y = element_blank(),
    axis.ticks.y = element_blank(),
    axis.title.y = element_blank(),
    strip.text.y = element_text(angle=0)) + scale_fill_manual(values = c(colour.Control, colour.Breast))

ED1d 

```

##  e) Cervical: Cell type boxplot BC / control.

```{r}
subset_data <- subset(data, set %in% c("discovery_cervical", "validation_cervical"))
subset_data$type <- factor(subset_data$type, levels=c("Control","BC case"))

# Define the new facet labels
labs <- c("Basophils", "CD4T", "CD8T", "Eosinophils", "Epithelial", "Fibroblasts", "Monocytes", "Neutrophils", "NK cells")
names(labs) <- c("hepidish_B", "hepidish_CD4T", "hepidish_CD8T", "hepidish_Eosino", "hepidish_Epi", "hepidish_Fib", "hepidish_Mono", "hepidish_Neutro", "hepidish_NK")

labs_names <- c("Discovery Set", "Validation Set")
names(labs_names) <- c("discovery_cervical", "validation_cervical")

# Define the columns you want to gather
celltypes <- c("hepidish_B", "hepidish_CD4T", "hepidish_CD8T", "hepidish_Eosino", "hepidish_Epi", "hepidish_Fib", "hepidish_Mono", "hepidish_Neutro", "hepidish_NK")

# Reshape the dataframe to long format
subset_data_long <- subset_data |>
  gather(key = "cell_type", value = "Fraction", celltypes)


ED1e <- ggplot(subset_data_long, aes(x = type, y = Fraction, fill = type)) +
  geom_boxplot(outlier.shape = NA) +
  coord_flip() +
  facet_grid(cell_type~set, labeller = labeller(cell_type= labs, set=labs_names))  +
  labs(y = "Proportion", x = "Cell type", fill = "", title= "Cervical") + 
  ggpubr::stat_compare_means(ref.group = 'Control',
                             label = 'p.signif',
                             hide.ns = T, label.y = c(0.9), vjust = +0.7) +
  theme_minimal() +
  theme(legend.position = "none",  # Move the legend to the bottom
        legend.direction = "horizontal",
        panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    axis.text.y = element_blank(),
    axis.ticks.y = element_blank(),
    axis.title.y = element_blank(),
    strip.text.y = element_text(angle=0)) + scale_fill_manual(values = c(colour.Control, colour.Breast))
ED1e
```

##  f) Blood: Cell type boxplot BC / control.

```{r}
subset_data <- subset(data, set %in% c("discovery_blood"))
subset_data2 <- blood_val
subset_data <- plyr::rbind.fill(subset_data, blood_val)
subset_data$type <- factor(subset_data$type, levels=c("Control","BC case"))

# Define the new facet labels
labs <- c("Basophils", "CD4T", "CD8T", "Eosinophils", "Epithelial", "Fibroblasts", "Monocytes", "Neutrophils", "NK cells")
names(labs) <- c("hepidish_B", "hepidish_CD4T", "hepidish_CD8T", "hepidish_Eosino", "hepidish_Epi", "hepidish_Fib", "hepidish_Mono", "hepidish_Neutro", "hepidish_NK")

labs_names <- c("Discovery Set", "Validation Set")
names(labs_names) <- c("discovery_blood", "validation_blood")

# Define the columns you want to gather
celltypes <- c("hepidish_B", "hepidish_CD4T", "hepidish_CD8T", "hepidish_Eosino", "hepidish_Epi", "hepidish_Fib", "hepidish_Mono", "hepidish_Neutro", "hepidish_NK")

# Reshape the dataframe to long format
subset_data_long <- subset_data |>
  gather(key = "cell_type", value = "Fraction", celltypes)


ED1f <- ggplot(subset_data_long, aes(x = type, y = Fraction, fill = type)) +
  geom_boxplot(outlier.shape = NA) +
  coord_flip() +
  facet_grid(cell_type~set, labeller = labeller(cell_type= labs, set=labs_names))  +
  labs(y = "Proportion", x = "Cell type", fill = "", title= "Blood") + 
  ggpubr::stat_compare_means(ref.group = 'Control',
                             label = 'p.signif',
                             hide.ns = T, label.y = c(0.9), vjust = +0.7) +
  theme_minimal() +
  theme(legend.position = "none",  # Move the legend to the bottom
        legend.direction = "vertical",
        panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    axis.text.y = element_blank(),
    axis.ticks.y = element_blank(),
    axis.title.y = element_blank(),
    strip.text.y = element_text(angle=0)) + scale_fill_manual(values = c(colour.Control, colour.Breast))
ED1f
```

## Compile and save

```{r, message=FALSE, echo=FALSE}
layout <- "
AABBCCD
EEEEEEE
FFFFFFF
GGGGGGG
"


pdf_width <- 8.3  # Width in inches (a4)
pdf_height <- 11.7   # Height in inches (a4: 11.7)


compiled <- ED1a + ED1b+ ED1c + guide_area() + ED1d+ ED1e+ ED1f
  
compiled <-  compiled + plot_annotation(tag_levels = 'a') + 
  plot_layout(design = layout, guides = 'collect') &
    theme(plot.tag = element_text(face = 'bold'))

ggsave(file.path(outPath,"ED3.pdf"), width = pdf_width, height = pdf_height, compiled)
# ggsave(file.path(jpgPath,"ED3.jpg"), width = pdf_width, height = pdf_height, compiled)
```




# Extended Data 2: PCA plot of all data

## a-c) PCA of all data

```{r}

scatter_size <- 0.4 # size of plotted points
legend_size <- 1.5 # size of points as displayed in legend
legend_text_size <- 0.7 # text size of legend entries, relative to base

load(here("0-data","dataframes-plotting","pca_all.Rdata"))

# Identify the indices where pcs$subset equals "GSE237036"
indices <- which(pcs$subset == "GSE237036")

# Update only those specific indices in pcs$ic
pcs$ic[indices] <- 1 - (pcs$epi[indices] + pcs$fib[indices])

# Recode some columns for plotting and consistency
pcs <- pcs |>
  mutate(subset = case_when(
    sampletype == 'buccal' ~ recode(subset,
                                     'external_validation' = 'Buccal validation set (EPIC)', 
                                     'brca' = 'BRCA1/2 mutation carrier set (EPIC)',
                                     'internal_validation' = 'Buccal Discovery Set (EPIC)',
                                     'training' = 'Buccal Discovery Set (EPIC)'),
    sampletype == 'cervical' ~ recode(subset,
                                       'external_validation' = 'Cervical validation set (EPIC)', 
                                       'brca' = 'BRCA1/2 mutation carrier set (EPIC)',
                                       'internal_validation' = 'Cervical Discovery Set (EPIC)',
                                       'training' = 'Cervical Discovery Set (EPIC)'),
    sampletype == 'blood' ~ recode(subset,
                                    'GSE237036' = 'Blood validation set (GSE237036 - EPIC)', 
                                    'brca' = 'BRCA1/2 mutation carrier set (EPIC)',
                                    'internal_validation' = 'Blood Discovery Set (EPIC)',
                                    'training' = 'Blood Discovery Set (EPIC)'),
    sampletype == 'breast' ~ recode(subset,
                                     'breast_tissue_epic' = 'Tissue at risk set (EPIC)', 
                                     'TCGA' = 'TCGA Set (450K)',
                                     'GSE225845' = 'GSE225845 (EPIC)'),
    TRUE ~ subset  # For all other cases, keep subset unchanged
  )) |> 
  mutate(type = case_when(
    sampletype == 'breast'~ recode(type, 'adj_norm'='Normal adjacent tissue',
                                 'Cancer'='Breast cancer tissue*',
                                 'Control'='Matched normal tissue*',
                                 'Normal'='Normal tissue',
                                 'Normal_adjacent'='Normal adjacent tissue',
                                 'Triple_negative_tumour'='Breast cancer tissue',
                                 'tumor'='Breast cancer tissue',
                                 'Risk_reduction_BRCA1'='Normal from BRCA1 mutation carrier',
                                 'Risk_reduction_BRCA2'='Normal from BRCA2 mutation carrier',
                                 'normal'='Normal tissue'),
    sampletype != 'breast' ~ recode(type, 'BRCA1'='Normal from BRCA1 mutation carrier',
                                 'BRCA2'='Normal from BRCA2 mutation carrier'),
    TRUE ~ type))

cols <- c('Buccal validation set (EPIC)' = "orange", 'BRCA1/2 mutation carrier set (EPIC)' = pal.jama[5], 'Buccal Discovery Set (EPIC)' = "lightgreen",
'Cervical validation set (EPIC)' = "darkblue", 'Cervical Discovery Set (EPIC)' = "lightblue",
'Blood validation set (GSE237036 - EPIC)' = "darkred", 'Blood Discovery Set (EPIC)' = "indianred", 
'Tissue at risk set (EPIC)'= "darkgrey", 'TCGA Set (450K)'= "lightgrey", 'GSE225845 (EPIC)'= "black")
breaks <- c('Buccal validation set (EPIC)', 'BRCA1/2 mutation carrier set (EPIC)', 'Buccal Discovery Set (EPIC)',
'Cervical validation set (EPIC)', 'Cervical Discovery Set (EPIC)',
'Blood validation set (GSE237036 - EPIC)', 'Blood Discovery Set (EPIC)', 
'Tissue at risk set (EPIC)', 'TCGA Set (450K)', 'GSE225845 (EPIC)')

cols2 <- c('Normal adjacent tissue'='lightsalmon','Breast cancer tissue*'='palevioletred','Matched normal tissue*'='cyan2','Normal tissue'=colour.Control,'Breast cancer tissue'=colour.Breast,'Normal from BRCA1 mutation carrier'="#79b245",'Normal from BRCA2 mutation carrier'="#2e752f", "BC case" = colour.Breast, "Control"= colour.Control)
breaks2 <- c('Normal adjacent tissue','Breast cancer tissue*','Matched normal tissue*','Normal tissue','Breast cancer tissue','Normal from BRCA1 mutation carrier','Normal from BRCA2 mutation carrier',"BC case","Control")


lab_pc1 <- paste0('<b>PC1</b> (', round(pcs$variance[1]*100), "%)")
lab_pc2 <- paste0('<b>PC2</b> (', round(pcs$variance[2]*100), "%)")
title_lab <- paste0("<span style = 'font-size:10pt;'><b>All samples</b> // IC</span></span>")
title_lab2 <- paste0("<span style = 'font-size:10pt;'>// Sample type</span></span>")
title_lab3 <- paste0("<span style = 'font-size:10pt;'>// Data set</span></span>")

# Plot
pca_all.ic <- ggplot(data = pcs, aes(x = PC1, y = PC2, color = ic)) +
  geom_point(alpha = 1, size = scatter_size) +  # Plot points
 # geom_rug(col=type,alpha=0.1, size=1.5) + 
  labs(
    y = "PC2",
    x = "PC1",
    color = "IC/neutrophil proportion"  # Rename color legend title
  ) +
  theme_bw() +
  theme(axis.title.y = element_markdown(),
        axis.title.x = element_markdown(),
        plot.title = element_markdown(),
        legend.text=element_text(size=rel(legend_text_size)),
        legend.title.align=1,
  legend.direction="vertical") +
  scale_shape_manual(values = c(1, 16, 13, 14, 10,11,12,15,2,3,4)) +
  scale_color_gradient(limits = c(0,1), low = "#00E3FF", high = "#FF1C00") + 
  labs(x = lab_pc1,
       y = lab_pc2,
       title=title_lab) + 
  guides(color = guide_colourbar(position = "left",title.position='top'))



pca_all.sample <- ggplot(data = pcs, aes(x = PC1, y = PC2, col=sampletype)) +
  geom_point(alpha = 1, size = scatter_size) +
 # geom_rug(col=type,alpha=0.1, size=1.5) + 
  labs(
    y = "PC2",
    x = "PC1",
    color = "Sample type",  # Rename color legend title
  ) +
 theme_bw() +
  theme(axis.title.y = element_markdown(),
        axis.title.x = element_markdown(),
        plot.title = element_markdown(),
        legend.text=element_text(size=rel(legend_text_size)),
        legend.title.align=1,
  legend.direction="vertical") +
  scale_color_manual(values = c("buccal" = colour.green, "cervical" = colour.blue, "blood" = colour.red,"breast" = "grey"))  + 
  labs(x = lab_pc1,
       y = lab_pc2,
       title=title_lab2) + 
  guides(color = guide_legend(nrow=5,override.aes = list(size = legend_size), label.position = "left",label.hjust = 1,title.position='top'))

pca_all.data <- ggplot(data = pcs, aes(x = PC1, y = PC2, col=subset)) +
  geom_point(alpha = 1, size = scatter_size) +
 # geom_rug(col=type,alpha=0.1, size=1.5) + 
  labs(
    y = "PC2",
    x = "PC1",
    color = "Data set"  # Rename color legend title
  ) + theme_bw() +
  theme(axis.title.y = element_markdown(),
        axis.title.x = element_markdown(),
        plot.title = element_markdown(),
        legend.text=element_text(size=rel(legend_text_size)),
        legend.title.align=1,
  legend.direction="vertical") +
  labs(x = lab_pc1,
       y = lab_pc2,
       title=title_lab3)  + scale_colour_manual(
  values = cols,
  breaks = breaks
) + 
  guides(color = guide_legend(nrow=length(unique(pcs$subset)),override.aes = list(size = legend_size), label.position = "left",label.hjust = 1,title.position='top'))




```


## d-f) PCA of all buccal data

```{r}
load(here("0-data","dataframes-plotting","pca_all_buccal.Rdata"))

# Recode some columns for plotting and consistency
pcs <- pcs |> mutate(subset=recode(subset, 'external_validation'='Buccal validation set (EPIC)', 
                             'brca'='BRCA1/2 mutation carrier set (EPIC)',
                             'internal_validation'='Buccal Discovery Set (EPIC)',
                             'training'='Buccal Discovery Set (EPIC)'), 
                      type= recode(type, 'BRCA1'='Normal from BRCA1 mutation carrier',
                                 'BRCA2'='Normal from BRCA2 mutation carrier'))

lab_pc1 <- paste0('<b>PC1</b> (', round(pcs$variance[1]*100), "%)")
lab_pc2 <- paste0('<b>PC2</b> (', round(pcs$variance[2]*100), "%)")
title_lab <- paste0("<span style = 'font-size:10pt;'><b>Buccal</b> // IC</span></span>")
title_lab2 <- paste0("<span style = 'font-size:10pt;'>// Type</span>")
title_lab3 <- paste0("<span style = 'font-size:10pt;'> // Data set</span>")

# Plot
pca_buccal.data <- ggplot(data = pcs, aes(x = PC1, y = PC2, col=subset)) +
  geom_point(alpha = 1, size = scatter_size) +
 # geom_rug(col=type,alpha=0.1, size=1.5) + 
  labs(
    y = "PC2",
    x = "PC1",
    color = "Data set"  # Rename color legend title
  ) + theme_bw() +
  theme(axis.title.y = element_markdown(),
        axis.title.x = element_markdown(),
        plot.title = element_markdown(),
  legend.position="none") +
  labs(x = lab_pc1,
       y = lab_pc2,
       title=title_lab3)  + scale_colour_manual(
  values = cols,
  breaks = breaks
)

# Plot
pca_buccal.ic <- ggplot(data = pcs, aes(x = PC1, y = PC2, color = ic)) +
  geom_point(alpha = 1, size = scatter_size) +  # Plot points
 # geom_rug(col=type,alpha=0.1, size=1.5) + 
  labs(
    y = "PC2",
    x = "PC1",
    color = "IC"  # Rename color legend title
  ) +
  theme_bw() +
  theme(axis.title.y = element_markdown(),
        axis.title.x = element_markdown(),
        plot.title = element_markdown(),
  legend.position="none") +
  scale_shape_manual(values = c(1, 16, 13, 14, 10,11,12,15,2,3,4)) +
  scale_color_gradient(limits = c(0,1), low = "#00E3FF", high = "#FF1C00") + 
  labs(x = lab_pc1,
       y = lab_pc2,
       title=title_lab) 

# Plot
pca_buccal.sample <- ggplot(data = pcs, aes(x = PC1, y = PC2, color = type)) +
  geom_point(alpha = 1, size = scatter_size) +  # Plot points
 # geom_rug(col=type,alpha=0.1, size=1.5) + 
  labs(
    y = "PC2",
    x = "PC1",
    color = "Type (buccal/cervical/blood)"  # Rename color legend title
  )+ theme_bw() +
  theme(axis.title.y = element_markdown(),
        axis.title.x = element_markdown(),
        plot.title = element_markdown(),
        legend.text=element_text(size=rel(legend_text_size)),
                legend.title.align=1,
  legend.direction="vertical") +
  labs(x = lab_pc1,
       y = lab_pc2,
       title=title_lab2)  + scale_colour_manual(
  values = cols2,
  breaks = breaks2
) + guides(color = guide_legend(nrow=5,override.aes = list(size = legend_size), label.position = "left",label.hjust = 1,title.position='top')) 

```

## g-i) PCA of all cervical data

```{r}
load(here("0-data","dataframes-plotting","pca_all_cervical.Rdata"))

# Recode some columns for plotting and consistency
pcs <- pcs |> mutate(subset=recode(subset, 'external_validation'='Cervical validation set (EPIC)', 
                             'brca'='BRCA1/2 mutation carrier set (EPIC)',
                             'internal_validation'='Cervical Discovery Set (EPIC)',
                             'training'='Cervical Discovery Set (EPIC)'), 
                      type= recode(type, 'BRCA1'='Normal from BRCA1 mutation carrier',
                                 'BRCA2'='Normal from BRCA2 mutation carrier'))



lab_pc1 <- paste0('<b>PC1</b> (', round(pcs$variance[1]*100), "%)")
lab_pc2 <- paste0('<b>PC2</b> (', round(pcs$variance[2]*100), "%)")
title_lab <- paste0("<span style = 'font-size:10pt;'><b>Cervical</b> // IC</span></span>")
title_lab2 <- paste0("<span style = 'font-size:10pt;'>// Type</span>")
title_lab3 <- paste0("<span style = 'font-size:10pt;'> // Data set</span>")

# Plot
pca_cervical.data <- ggplot(data = pcs, aes(x = PC1, y = PC2, col=subset)) +
  geom_point(alpha = 1, size = scatter_size) +
 # geom_rug(col=type,alpha=0.1, size=1.5) + 
  labs(
    y = "PC2",
    x = "PC1",
    color = "Data set"  # Rename color legend title
  ) + theme_bw() +
  theme(axis.title.y = element_markdown(),
        axis.title.x = element_markdown(),
        plot.title = element_markdown(),
  legend.position="none") +
  labs(x = lab_pc1,
       y = lab_pc2,
       title=title_lab3)  + scale_colour_manual(
  values = cols,
  breaks = breaks
)

# Plot
pca_cervical.ic <- ggplot(data = pcs, aes(x = PC1, y = PC2, color = ic)) +
  geom_point(alpha = 1, size = scatter_size) +  # Plot points
 # geom_rug(col=type,alpha=0.1, size=1.5) + 
  labs(
    y = "PC2",
    x = "PC1",
    color = "IC"  # Rename color legend title
  ) +
  theme_bw() +
  theme(axis.title.y = element_markdown(),
        axis.title.x = element_markdown(),
        plot.title = element_markdown(),
  legend.position="none") +
  scale_shape_manual(values = c(1, 16, 13, 14, 10,11,12,15,2,3,4)) +
  scale_color_gradient(limits = c(0,1), low = "#00E3FF", high = "#FF1C00") + 
  labs(x = lab_pc1,
       y = lab_pc2,
       title=title_lab) 

# Plot
pca_cervical.sample <- ggplot(data = pcs, aes(x = PC1, y = PC2, color = type)) +
  geom_point(alpha = 1, size = scatter_size) +  # Plot points
 # geom_rug(col=type,alpha=0.1, size=1.5) + 
  labs(
    y = "PC2",
    x = "PC1",
    color = "Type"  # Rename color legend title
  )+ theme_bw() +
  theme(axis.title.y = element_markdown(),
        axis.title.x = element_markdown(),
        plot.title = element_markdown(),
        legend.position="none") +
  labs(x = lab_pc1,
       y = lab_pc2,
       title=title_lab2)  + scale_colour_manual(
  values = cols2,
  breaks = breaks2
)

```


## j-l) PCA of all blood data

```{r}
load(here("0-data","dataframes-plotting","pca_all_blood.Rdata"))

# Identify the indices where pcs$subset equals "GSE237036"
indices <- which(pcs$subset == "GSE237036")

# Update only those specific indices in pcs$ic
pcs$ic[indices] <- 1 - (pcs$epi[indices] + pcs$fib[indices])

# Recode some columns for plotting and consistency
pcs <- pcs |> mutate(subset=recode(subset, 'GSE237036'='Blood validation set (GSE237036 - EPIC)', 
                             'brca'='BRCA1/2 mutation carrier set (EPIC)',
                             'internal_validation'='Blood Discovery Set (EPIC)',
                             'training'='Blood Discovery Set (EPIC)'), 
                      type= recode(type, 'BRCA1'='Normal from BRCA1 mutation carrier',
                                 'BRCA2'='Normal from BRCA2 mutation carrier'))



lab_pc1 <- paste0('<b>PC1</b> (', round(pcs$variance[1]*100), "%)")
lab_pc2 <- paste0('<b>PC2</b> (', round(pcs$variance[2]*100), "%)")
title_lab <- paste0("<span style = 'font-size:10pt;'><b>Blood</b> // Neutrophil</span></span>")
title_lab2 <- paste0("<span style = 'font-size:10pt;'>// Type</span>")
title_lab3 <- paste0("<span style = 'font-size:10pt;'> // Data set</span>")

# Plot
pca_blood.data <- ggplot(data = pcs, aes(x = PC1, y = PC2, col=subset)) +
  geom_point(alpha = 1, size = scatter_size) +
 # geom_rug(col=type,alpha=0.1, size=1.5) + 
  labs(
    y = "PC2",
    x = "PC1",
    color = "Data set"  # Rename color legend title
  ) + theme_bw() +
  theme(axis.title.y = element_markdown(),
        axis.title.x = element_markdown(),
        plot.title = element_markdown(),
  legend.position="none") +
  labs(x = lab_pc1,
       y = lab_pc2,
       title=title_lab3)  + scale_colour_manual(
  values = cols,
  breaks = breaks
)

# Plot
pca_blood.ic <- ggplot(data = pcs, aes(x = PC1, y = PC2, color = hepidish_Neutro)) +
  geom_point(alpha = 1, size = scatter_size) +  # Plot points
 # geom_rug(col=type,alpha=0.1, size=1.5) + 
  labs(
    y = "PC2",
    x = "PC1",
    color = "Neutrophil"  # Rename color legend title
  ) +
  theme_bw() +
  theme(axis.title.y = element_markdown(),
        axis.title.x = element_markdown(),
        plot.title = element_markdown(),
  legend.position="none") +
  scale_shape_manual(values = c(1, 16, 13, 14, 10,11,12,15,2,3,4))  + scale_color_gradient(limits = c(0,1), low = "#00E3FF", high = "#FF1C00") + 
  labs(x = lab_pc1,
       y = lab_pc2,
       title=title_lab) 

# Plot
pca_blood.sample <- ggplot(data = pcs, aes(x = PC1, y = PC2, color = type)) +
  geom_point(alpha = 1, size = scatter_size) +  # Plot points
 # geom_rug(col=type,alpha=0.1, size=1.5) + 
  labs(
    y = "PC2",
    x = "PC1",
    color = "Type"  # Rename color legend title
  )+ theme_bw() +
  theme(axis.title.y = element_markdown(),
        axis.title.x = element_markdown(),
        plot.title = element_markdown(),
  legend.position="none") +
  labs(x = lab_pc1,
       y = lab_pc2,
       title=title_lab2)  + scale_colour_manual(
  values = cols2,
  breaks = breaks2
)

```



## m-o) PCA of all breast data

```{r}
load(here("0-data","dataframes-plotting","pca_all_breast.Rdata"))

# Recode some columns for plotting and consistency
pcs <- pcs |> mutate(type=recode(type, 'adj_norm'='Normal adjacent tissue',
                                 'Cancer'='Breast cancer tissue*',
                                 'Control'='Matched normal tissue*',
                                 'Normal'='Normal tissue',
                                 'Normal_adjacent'='Normal adjacent tissue',
                                 'Triple_negative_tumour'='Breast cancer tissue',
                                 'tumor'='Breast cancer tissue',
                                 'Risk_reduction_BRCA1'='Normal from BRCA1 mutation carrier',
                                 'Risk_reduction_BRCA2'='Normal from BRCA2 mutation carrier',
                                 'normal'='Normal tissue'),
               subset=recode(subset, 'breast_tissue_epic'='Tissue at risk set (EPIC)', 
                             'TCGA'='TCGA Set (450K)',
                             'GSE225845'='GSE225845 (EPIC)'))



lab_pc1 <- paste0('<b>PC1</b> (', round(pcs$variance[1]*100), "%)")
lab_pc2 <- paste0('<b>PC2</b> (', round(pcs$variance[2]*100), "%)")
title_lab <- paste0("<span style = 'font-size:10pt;'><b>Breast</b> // IC</span></span>")
title_lab2 <- paste0("<span style = 'font-size:10pt;'>// Type</span>")
title_lab3 <- paste0("<span style = 'font-size:10pt;'> // Data set</span>")

# Plot
pca_breast.data <- ggplot(data = pcs, aes(x = PC1, y = PC2, col=subset)) +
  geom_point(alpha = 1, size = scatter_size) +
 # geom_rug(col=type,alpha=0.1, size=1.5) + 
  labs(
    y = "PC2",
    x = "PC1",
    color = "Data set"  # Rename color legend title
  ) + theme_bw() +
  theme(axis.title.y = element_markdown(),
        axis.title.x = element_markdown(),
        plot.title = element_markdown(),
  legend.position="none") +
  labs(x = lab_pc1,
       y = lab_pc2,
       title=title_lab3)  + scale_colour_manual(
  values = cols,
  breaks = breaks
)

# Plot
pca_breast.ic <- ggplot(data = pcs, aes(x = PC1, y = PC2, color = ic)) +
  geom_point(alpha = 1, size = scatter_size) +  # Plot points
 # geom_rug(col=type,alpha=0.1, size=1.5) + 
  labs(
    y = "PC2",
    x = "PC1",
    color = "IC"  # Rename color legend title
  ) +
  theme_bw() +
  theme(axis.title.y = element_markdown(),
        axis.title.x = element_markdown(),
        plot.title = element_markdown(),
  legend.position="none") +
  scale_shape_manual(values = c(1, 16, 13, 14, 10,11,12,15,2,3,4)) +
  scale_color_gradient(limits = c(0,1), low = "#00E3FF", high = "#FF1C00") + 
  labs(x = lab_pc1,
       y = lab_pc2,
       title=title_lab) 

# Plot
pca_breast.sample <- ggplot(data = pcs, aes(x = PC1, y = PC2, color = type)) +
  geom_point(alpha = 1, size = scatter_size) +  # Plot points
 # geom_rug(col=type,alpha=0.1, size=1.5) + 
  labs(
    y = "PC2",
    x = "PC1",
    color = "Type (breast)"  # Rename color legend title
  )+ theme_bw() +
  theme(axis.title.y = element_markdown(),
        axis.title.x = element_markdown(),
        plot.title = element_markdown(),
                legend.title.align=1,
        legend.text=element_text(size=rel(legend_text_size)),
  legend.direction="vertical") +
  labs(x = lab_pc1,
       y = lab_pc2,
       title=title_lab2)  + scale_colour_manual(
  values = cols2,
  breaks = breaks2
) + 
  guides(color = guide_legend(nrow=length(unique(pcs$type)),override.aes = list(size = legend_size), label.position = "left",label.hjust = 1,title.position='top'))



```

## Compile and save

```{r}



layout <- "
BBCCDDAAA
EEFFGGAAA
HHIIJJAAA
KKLLMMAAA
NNOOPPAAA
"



pdf_width <- 8.3   # Width in inches (a4)
pdf_height <- 11.7  # Height in inches (a4: 11.7)



compiled <- guide_area() + pca_all.ic+pca_all.sample+pca_all.data + pca_buccal.ic+pca_buccal.sample+pca_buccal.data + pca_cervical.ic+pca_cervical.sample+pca_cervical.data + pca_blood.ic+pca_blood.sample+pca_blood.data + pca_breast.ic+pca_breast.sample+pca_breast.data 
compiled <-  compiled + plot_annotation(tag_levels = 'a')  +
  plot_layout(design = layout, guides = "collect", axes = "collect")



ggsave(file.path(outPath,"EDx.pdf"), width = pdf_width, height = pdf_height, compiled)


```





# Extended Data 3: P value and manhattan plots

Manhattan plots take a long time to generate, hence they've been generated already and saved as .jpgs. These can be used if the figure needs to be re-arranged. They are called from the 'compile' section below.

## a) Buccal: P value histogram 

```{r}
buccal <- readRDS(here("1-ewas", "1-buccal", "delta-beta.Rds")) |> 
  as.data.frame() |> 
  dplyr::mutate(padj = p.adjust(type, method = 'BH')) |> 
  tibble::rownames_to_column('cg')




# Create a histogram 
s12a <- ggplot(buccal, aes(x=type)) +
  geom_histogram(binwidth=0.005, fill=pal.jama[1]) +
  labs(x = "P-value",
       y = "Frequency",
       title = 'Buccal Discovery Set') +
  theme_minimal() +
  theme(
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank()
  )   

s12a
```
## b) Manhattan plot buccal DMPs

Uncomment to re-generate the manhattan plot (takes long, and files have been saved; 'compile' section below.)

```{r}
# rownames(buccal) <- buccal$cg
# buccal$cg <- NULL


# mh_buccal <- manhattan(buccal, label_cutoff = 7)
# 
# save(mh_buccal, file=file.path(sepPath,"manhattan-buccal.Rdata"))
# ggsave(plot = c1,
#       file.path(sepPath,"manhattan-buccal.jpg"), width = 10, height = 4)
```



## b) Cervical: P value histogram

```{r}
cervical <- readRDS(here("1-ewas", "2-cervical", "delta-beta.Rds")) |> 
  as.data.frame() |> 
  dplyr::mutate(padj = p.adjust(type, method = 'BH')) |> 
  tibble::rownames_to_column('cg')

# Create a histogram 
s12b <- ggplot(cervical, aes(x=type)) +
  geom_histogram(binwidth=0.005, fill=pal.jama[1]) +
  labs(x = "P-value",
       y = "Frequency",
       title = 'Cervical Discovery Set') +
  theme_minimal() +
  theme(
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank()
  )

s12b
```
## b) Manhattan plot cervical DMPs

Uncomment to re-generate the manhattan plot (takes long, and files have been saved; 'compile' section below.)

```{r}
# rownames(cervical) <- cervical$cg
# cervical$cg <- NULL
# 
# 
# mh_cervical <- manhattan(cervical, label_cutoff = 7)
# 
# save(mh_cervical, file=file.path(sepPath,"manhattan-cervical.Rdata"))

```

## b) Blood: P value histogram

```{r}
blood <- readRDS(here("1-ewas", "3-blood", "delta-beta.Rds")) |> 
  as.data.frame() |> 
  dplyr::mutate(padj = p.adjust(type, method = 'BH')) |> 
  tibble::rownames_to_column('cg')

# Create a histogram 
s12c <- ggplot(blood, aes(x=type)) +
  geom_histogram(binwidth=0.005, fill=pal.jama[1]) +
  labs(x = "P-value",
       y = "Frequency",
       title = 'Blood Discovery Set') +
  theme_minimal() +
  theme(
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank()
  )

s12c
```
## Manhattan plot blood

Uncomment to re-generate the manhattan plot (takes long, and files have been saved; 'compile' section below.)

```{r}
# rownames(blood) <- blood$cg
# blood$cg <- NULL
# 
# 
# mh_blood <- manhattan(blood, label_cutoff = 7)
# 
# save(mh_blood, file=file.path(sepPath,"manhattan-blood.Rdata"))
```



## Compile and save

```{r, message=FALSE, echo=FALSE}

layout <- "
AB
CD
EF
"
source(file.path(sepPath,"manhattan-buccal.Rdata"))
source(file.path(sepPath,"manhattan-cervical.Rdata"))
source(file.path(sepPath,"manhattan-blood.Rdata"))

pdf_width <- 8.3  # Width in inches (a4)
pdf_height <- 8   # Height in inches (a4: 11.7)


compiled <- s12a + mh_buccal + s12b + mh_cervical + s12c + mh_blood
  
compiled <-  compiled + plot_annotation(tag_levels = 'a') + 
  plot_layout(design = layout)

ggsave(file.path(outPath,"ED4.pdf"), width = pdf_width, height = pdf_height, compiled)
ggsave(file.path(jpgPath,"ED4.jpg"), width = pdf_width, height = pdf_height, compiled)
```




# Extended Data 4: Relation to Island

## a) Buccal

```{r}
EDxa <-pie_chart_dmps("ST3.csv")


fileName <- "ST3.csv"
supPath < here("5-supplements")


colour.pal.d8 <- c("#EA7580","#F2949A","#F6B3A1","#D8C99E","#3AB9AC","#109DB7","#0C6EA5","#172869")

  colour.pal.c <- colorRampPalette(colour.pal.d8)
  
  # Read table from supplements folder
  df <- read.csv(file.path(supPath, fileName))
  
  # 1. Select entries with padj < 0.05
  selected_entries <- df[df$padj < 0.05, ]
  
  # Reformat the multiple names to first value
  selected_entries$UCSC_RefGene_Group<- stringr::str_split(  selected_entries$UCSC_RefGene_Group, ";", simplify = T)[,1]
  

  selected_entries$UCSC_RefGene_Group <-   ifelse(selected_entries$UCSC_RefGene_Group == "" | is.na(selected_entries$UCSC_RefGene_Group), "Unknown", selected_entries$UCSC_RefGene_Group)

  
  sets <- length(unique(selected_entries$UCSC_RefGene_Group))
  
  # 3. Create pie chart
EDxb <- ggplot(selected_entries, aes(x = "", fill = UCSC_RefGene_Group)) +
  geom_bar(stat = "count", width = 1) +
  coord_polar("y") +
  scale_fill_manual(values = colour.pal.d8) +  # Specify custom colors
  labs(fill = "Gene group", x = NULL, y = NULL) +
  theme_void()


```


## b) Cervical

```{r}
EDxc <-pie_chart_dmps("ST4.csv")

fileName <- "ST4.csv"
supPath < here("5-supplements")


colour.pal.d8 <- c("#EA7580","#F2949A","#F6B3A1","#D8C99E","#3AB9AC","#109DB7","#0C6EA5","#172869")

  colour.pal.c <- colorRampPalette(colour.pal.d8)
  
  # Read table from supplements folder
  df <- read.csv(file.path(supPath, fileName))
  
  # 1. Select entries with padj < 0.05
  selected_entries <- df[df$padj < 0.05, ]
  
  # Reformat the multiple names to first value
  selected_entries$UCSC_RefGene_Group<- stringr::str_split(  selected_entries$UCSC_RefGene_Group, ";", simplify = T)[,1]
  

  selected_entries$UCSC_RefGene_Group <-   ifelse(selected_entries$UCSC_RefGene_Group == "" | is.na(selected_entries$UCSC_RefGene_Group), "Unknown", selected_entries$UCSC_RefGene_Group)

  
  sets <- length(unique(selected_entries$UCSC_RefGene_Group))
  
  # 3. Create pie chart
EDxd <- ggplot(selected_entries, aes(x = "", fill = UCSC_RefGene_Group)) +
  geom_bar(stat = "count", width = 1) +
  coord_polar("y") +
  scale_fill_manual(values = colour.pal.d8) +  # Specify custom colors
  labs(fill = "Gene group", x = NULL, y = NULL) +
  theme_void()
```




## Compile and save

```{r, message=FALSE, echo=FALSE}

layout <- "
AB
CD
"


pdf_width <- 8.3  # Width in inches (a4)
pdf_height <- 5   # Height in inches (a4: 11.7)


compiled <- (EDxa + theme(legend.position = "none") + labs(title="Buccal samples")) + (EDxc  + labs(title="Cervical samples")) + EDxb +  theme(legend.position = "none") + EDxd  
  
  
  
compiled <-  compiled + plot_annotation(tag_levels = 'a') + 
  plot_layout(design = layout) &
    theme(plot.tag = element_text(face = 'bold'))

ggsave(file.path(outPath,"ED5.pdf"), width = pdf_width, height = pdf_height, compiled)
# ggsave(file.path(jpgPath,"ED5.jpg"), width = pdf_width, height = pdf_height, compiled)
```



# Extended Data 5 

Final file has been compiled in Adobe Illustrator, hence do not regenerate unless data changes. Edit ED6.ai or ED6.pdf in illustrator directly.

## a + b) Hexbins in inferred epi/immune and buccal/cervical

```{r}
here::i_am("4-markdown-figures/or_enrichment_temp.R")

# Load DMPs
buccal <- readRDS(here("1-ewas", "1-buccal", "delta-beta.Rds")) |> 
  as.data.frame() |> 
  dplyr::mutate(padj = p.adjust(type, method = 'BH')) |> 
  tibble::rownames_to_column('cg')

cervical <- readRDS(here("1-ewas", "2-cervical", "delta-beta.Rds")) |> 
  as.data.frame() |> 
  dplyr::mutate(padj = p.adjust(type, method = 'BH'))|> 
  tibble::rownames_to_column('cg') 

# Merge
dmps <- buccal |> 
  dplyr::full_join(cervical, by = 'cg') |> 
  dplyr::filter(type.y < 0.05 & type.x < 0.05) # Keep those significant in both

# Odds ratios
ORquadrant <- function(p, x, n){
  
  # following functions here:
  # https://pages.stat.wisc.edu/~st571-1/06-tables-4.pdf
  p0 = 0.25 # Null hypothesis is equal distribution - 0.25
  or <- ((p/(1-p))/(p0/((1-p0)))) # compute OR
  
  x2 <- round(n/4)
  se <- sqrt ( (1/x) + (1/(n-x)) + (1/x2) + (1/(n-x2)) )
  
  
  df <- data.frame(or = or,
                   ci_lo = or - 1.96*se,
                   ci_hi = or + 1.96*se)
  return(df)
  
}

# Plot: DELTA BETA EPITHELIAL direction -----------------------------------
# BuccalHypo, CervHyper
p = nrow(dmps[dmps$db_epithelial.x<0 & dmps$db_epithelial.y>0,])/nrow(dmps)
x = nrow(dmps[dmps$db_epithelial.x<0 & dmps$db_epithelial.y>0,])

BuccalHypo_CervHyper <- ORquadrant(p, x, n = nrow(dmps))
anno_BHoCHer <- paste0('Buccal <b style="color:#0000FF">hypo</b><br>Cervical <b style="color:#FF0000">hyper</b><br>OR = ', format(round(BuccalHypo_CervHyper$or, 2), nsmall = 2), '<br>[', 
                        format(round(BuccalHypo_CervHyper$ci_lo, 2), nsmall = 2),
                        ' - ',
                        format(round(BuccalHypo_CervHyper$ci_hi, 2), nsmall = 2), ']')

# BuccalHyper, CervHyper
p = nrow(dmps[dmps$db_epithelial.x>0 & dmps$db_epithelial.y>0,])/nrow(dmps)
x = nrow(dmps[dmps$db_epithelial.x>0 & dmps$db_epithelial.y>0,])

BuccalHyper_CervHyper <- ORquadrant(p, x, n = nrow(dmps))
BuccalHyper_CervHyper
anno_BHerCHer <- paste0('Buccal <b style="color:#FF0000">hyper</b><br>Cervical <b style="color:#FF0000">hyper</b><br>OR = ', format(round(BuccalHyper_CervHyper$or, 2), nsmall = 2), '<br>[', 
                        format(round(BuccalHyper_CervHyper$ci_lo, 2), nsmall = 2),
                        ' - ',
                        format(round(BuccalHyper_CervHyper$ci_hi, 2), nsmall = 2), ']')

# BuccalHypo, CervHypo
p = nrow(dmps[dmps$db_epithelial.x<0 & dmps$db_epithelial.y<0,])/nrow(dmps)
x = nrow(dmps[dmps$db_epithelial.x<0 & dmps$db_epithelial.y<0,])

BuccalHypo_CervHypo <- ORquadrant(p, x, n = nrow(dmps))
anno_BHoCHo <- paste0('Buccal <b style="color:#0000FF">hypo</b><br>Cervical <b style="color:#0000FF">hypo</b><br>OR = ', format(round(BuccalHypo_CervHypo$or, 2), nsmall = 2), '<br>[', 
                       format(round(BuccalHypo_CervHypo$ci_lo, 2), nsmall = 2),
                       ' - ',
                       format(round(BuccalHypo_CervHypo$ci_hi, 2), nsmall = 2), ']')

# BuccalHyper, CervHypo
p = nrow(dmps[dmps$db_epithelial.x>0 & dmps$db_epithelial.y<0,])/nrow(dmps)
x = nrow(dmps[dmps$db_epithelial.x>0 & dmps$db_epithelial.y<0,])

BuccalHyper_CervHypo <- ORquadrant(p, x, n = nrow(dmps))
anno_BHerCHo <- paste0('Buccal <b style="color:#FF0000">hyper</b><br>Cervical <b style="color:#0000FF">hypo</b><br>OR = ', format(round(BuccalHyper_CervHypo$or, 2), nsmall = 2), '<br>[', 
                      format(round(BuccalHyper_CervHypo$ci_lo, 2), nsmall = 2),
                      ' - ',
                      format(round(BuccalHyper_CervHypo$ci_hi, 2), nsmall = 2), ']')

epi <- dmps |> 
  dplyr::filter(type.y < 0.05 & type.x < 0.05) |> 
  ggplot(aes(x = db_epithelial.x,
             y = db_epithelial.y)) +
  geom_hex(bins = 100) +
  labs(x = 'Buccal epithelial Δβ',
       y = 'Cervical epithelial Δβ') +
  scale_fill_viridis_c(name = "Number of CpGs", option = "plasma") +
  ggpubr::stat_cor(method = "pearson", size = 3) + # Add correlation coefficient
  theme_minimal() + # Add this line to remove the grey background
  theme(legend.position = "bottom", legend.box = "horizontal") + # Move legend below and set it to horizontal
  annotate('richtext',
           x = 0.05, y = 0.07,
           label = anno_BHerCHer,
           label.size = 0.2,
           alpha = 0.8,
           hjust = 0
           )  +
  annotate('richtext',
           x = -0.05, y = 0.07,
           label = anno_BHoCHer,
           label.size = 0.2,
           alpha = 0.8,
           hjust = 1) +
  annotate('richtext',
           x = -0.05, y = -0.07,
           label = anno_BHoCHo,
           label.size = 0.2,
           alpha = 0.8,
           hjust = 1,
           vjust = 1) +
  annotate('richtext',
           x = 0.05, y = -0.07,
           label = anno_BHerCHo,
           label.size = 0.2,
           alpha = 0.8,
           hjust = 0,
           vjust = 1) +
  geom_vline(xintercept = 0,
             alpha = 0.6) +
  geom_hline(yintercept = 0,
             alpha = 0.6)


# Plot: DELTA BETA IMMUNE direction -----------------------------------
# BuccalHypo, CervHyper
p = nrow(dmps[dmps$db_immune.x<0 & dmps$db_immune.y>0,])/nrow(dmps)
x = nrow(dmps[dmps$db_immune.x<0 & dmps$db_immune.y>0,])

BuccalHypo_CervHyper <- ORquadrant(p, x, n = nrow(dmps))
anno_BHoCHer <- paste0('Buccal <b style="color:#0000FF">hypo</b><br>Cervical <b style="color:#FF0000">hyper</b><br>OR = ', format(round(BuccalHypo_CervHyper$or, 2), nsmall = 2), '<br>[', 
                       format(round(BuccalHypo_CervHyper$ci_lo, 2), nsmall = 2),
                       ' - ',
                       format(round(BuccalHypo_CervHyper$ci_hi, 2), nsmall = 2), ']')

# BuccalHyper, CervHyper
p = nrow(dmps[dmps$db_immune.x>0 & dmps$db_immune.y>0,])/nrow(dmps)
x = nrow(dmps[dmps$db_immune.x>0 & dmps$db_immune.y>0,])

BuccalHyper_CervHyper <- ORquadrant(p, x, n = nrow(dmps))
BuccalHyper_CervHyper
anno_BHerCHer <- paste0('Buccal <b style="color:#FF0000">hyper</b><br>Cervical <b style="color:#FF0000">hyper</b><br>OR = ', format(round(BuccalHyper_CervHyper$or, 2), nsmall = 2), '<br>[', 
                        format(round(BuccalHyper_CervHyper$ci_lo, 2), nsmall = 2),
                        ' - ',
                        format(round(BuccalHyper_CervHyper$ci_hi, 2), nsmall = 2), ']')

# BuccalHypo, CervHypo
p = nrow(dmps[dmps$db_immune.x<0 & dmps$db_immune.y<0,])/nrow(dmps)
x = nrow(dmps[dmps$db_immune.x<0 & dmps$db_immune.y<0,])

BuccalHypo_CervHypo <- ORquadrant(p, x, n = nrow(dmps))
anno_BHoCHo <- paste0('Buccal <b style="color:#0000FF">hypo</b><br>Cervical <b style="color:#0000FF">hypo</b><br>OR = ', format(round(BuccalHypo_CervHypo$or, 2), nsmall = 2), '<br>[', 
                      format(round(BuccalHypo_CervHypo$ci_lo, 2), nsmall = 2),
                      ' - ',
                      format(round(BuccalHypo_CervHypo$ci_hi, 2), nsmall = 2), ']')

# BuccalHyper, CervHypo
p = nrow(dmps[dmps$db_immune.x>0 & dmps$db_immune.y<0,])/nrow(dmps)
x = nrow(dmps[dmps$db_immune.x>0 & dmps$db_immune.y<0,])

BuccalHyper_CervHypo <- ORquadrant(p, x, n = nrow(dmps))
anno_BHerCHo <- paste0('Buccal <b style="color:#FF0000">hyper</b><br>Cervical <b style="color:#0000FF">hypo</b><br>OR = ', format(round(BuccalHyper_CervHypo$or, 2), nsmall = 2), '<br>[', 
                       format(round(BuccalHyper_CervHypo$ci_lo, 2), nsmall = 2),
                       ' - ',
                       format(round(BuccalHyper_CervHypo$ci_hi, 2), nsmall = 2), ']')


immune <- dmps |> 
  dplyr::filter(type.y < 0.05 & type.x < 0.05) |> 
  ggplot(aes(x = db_immune.x,
             y = db_immune.y)) +
  geom_hex(bins = 100) +
  labs(x = 'Buccal immune Δβ',
       y = 'Cervical immune Δβ') +
  scale_fill_viridis_c(name = "Number of CpGs", option = "plasma") +
  ggpubr::stat_cor(method = "pearson",size = 3) + # Add correlation coefficient
  theme_minimal() + # Add this line to remove the grey background
  theme(legend.position = "bottom", legend.box = "horizontal") + # Move legend below and set it to horizontal
  annotate('richtext',
           x = 0.05, y = 0.07,
           label = anno_BHerCHer,
           label.size = 0.2,
           alpha = 0.8,
           hjust = 0
  )  +
  annotate('richtext',
           x = -0.05, y = 0.07,
           label = anno_BHoCHer,
           label.size = 0.2,
           alpha = 0.8,
           hjust = 1) +
  annotate('richtext',
           x = -0.05, y = -0.07,
           label = anno_BHoCHo,
           label.size = 0.2,
           alpha = 0.8,
           hjust = 1,
           vjust = 1) +
  annotate('richtext',
           x = 0.05, y = -0.07,
           label = anno_BHerCHo,
           label.size = 0.2,
           alpha = 0.8,
           hjust = 0,
           vjust = 1) +
  geom_vline(xintercept = 0,
             alpha = 0.6) +
  geom_hline(yintercept = 0,
             alpha = 0.6)


```



## b) DMR density plot
Uncomment to regenerate the plot, otherwise load the plot from .jpg.

```{r}



# # Load data
# load(here("1-ewas", "1-buccal", "2-output", "ranges.Rdata")) 
# buccal <- ranges
# 
# load(here("1-ewas", "2-cervical", "2-output", "ranges.Rdata")) 
# cervical <- ranges
# 
# buccal <- sortSeqlevels(buccal)
# buccal <- sort(buccal)
# # buccal <- buccal[buccal$Stouffer<0.05,] # let's keep all significant ones in the list
# 
# cervical <- sortSeqlevels(cervical)
# cervical <- sort(cervical)
# # cervical <- cervical[cervical$Stouffer<0.05,]
# 
# # set up data
# buccal_down <- buccal[sign(buccal$meandiff) == -1,]
# buccal_up <- buccal[sign(buccal$meandiff) == 1,]
# 
# cervical_down <- cervical[sign(cervical$meandiff) == -1,]
# cervical_up <- cervical[sign(cervical$meandiff) == 1,]
# 
# window <- 6e6
# 
# pdf(file.path(sepPath,"b1.pdf"))
# 
# 
# kp <- plotKaryotype(plot.type=4,genome = 'hg19',
#                     chromosomes = paste0('chr', 1:22),
#                     labels.plotter = NULL,cex = 0.5,
#                     ideogram.plotter = NULL)
# kpAddChromosomeNames(kp, paste0(1:22), cex = 0.8)
# # kpDataBackground(kp, data.panel = 1)
# 
# kp <- kpPlotDensity(kp, buccal_up, window.size = window, col=transparent("#3388FF", 0.6), border=transparent("#3388FF", 0.6), r0=0.5, r1=1)
# kp <- kpPlotDensity(kp, buccal_down, window.size = window,col=transparent("#3388FF", 0.6), border=transparent("#3388FF", 0.6), r0=0.5, r1=0)
# 
# kp <- kpPlotDensity(kp, cervical_up, window.size = window,  col=transparent("red", 0.6), border=transparent("red", 0.6), r0=0.5, r1=1)
# kp <- kpPlotDensity(kp, cervical_down, window.size = window, col=transparent("red", 0.6), border=transparent("red", 0.6), r0=0.5, r1=0)
# 
# kpAxis(kp, ymax=1, ymin=-1,labels = c('hypoM', '0', 'hyperM'), cex = 0.5)
# 
# kpAddLabels(kp, labels = "DMR (density)", srt=90, pos = 1, label.margin = 0.04,
#             cex = 0.8)
# 
# dev.off()

# Import the image
img <- jpeg::readJPEG(file.path(sepPath,"b1.jpg"), native = TRUE)

ratio = 4447968 /16172005

b1 <- ggplot() +
  background_image(img) + theme(aspect.ratio = ratio) 

```


## c) example opposing DMRs

```{r}
# Load data
load(here("1-ewas", "1-buccal", "2-output", "ranges.Rdata")) 
buccal <- ranges

load(here("1-ewas", "2-cervical", "2-output", "ranges.Rdata")) 
cervical <- ranges

buccal <- sortSeqlevels(buccal)
buccal <- sort(buccal)
# buccal <- buccal[buccal$Stouffer<0.05,]

cervical <- sortSeqlevels(cervical)
cervical <- sort(cervical)
# cervical <- cervical[cervical$Stouffer,]

overlaps1 <- subsetByOverlaps(buccal, cervical)
overlaps2 <- subsetByOverlaps(cervical, buccal)

# Load pheno + data for buccal
load(here("1-ewas", "1-buccal", "pheno.Rdata"))
pheno_buccal <- pheno |>  dplyr::mutate(type = factor(type, levels = c("Control", "BC case")))
load(here(dbPath, "data", "3c-buccal", "beta_merged.Rdata"))
beta_buccal <- beta_merged[,rownames(pheno)]

# Load pheno + data for cervical
load(here("1-ewas", "2-cervical", "pheno.Rdata"))
load(here(dbPath, "data", "3c", "beta_merged.Rdata"))
pheno_cervical <- pheno |> dplyr::mutate(type = factor(type, levels = c("Control", "BC case")))
beta_cervical <- beta_merged[,rownames(pheno)]

source(here('0-data', "src", "plotDMR_comparison.R"))

# RP11
rp11_1 <- overlaps1[grepl('RP11-551', overlaps1$overlapping.genes) & !is.na(overlaps1$overlapping.genes),]
rp11_2 <- overlaps2[grepl('RP11-551', overlaps2$overlapping.genes) & !is.na(overlaps2$overlapping.genes),]
rp11 <- plotDMR_comparison(rp11_1, rp11_2, 
                           beta_buccal, beta_cervical,
                           pheno_buccal, pheno_cervical)

b <- overlaps1[grepl('LTBP4', overlaps1$overlapping.genes) & !is.na(overlaps1$overlapping.genes),]
a <- overlaps2[grepl('LTBP4', overlaps2$overlapping.genes) & !is.na(overlaps2$overlapping.genes),]
ltbp4 <- plotDMR_comparison(a, b, 
                           beta_buccal, beta_cervical,
                           pheno_buccal, pheno_cervical)

b <- overlaps1[grepl('NCK1', overlaps1$overlapping.genes) & !is.na(overlaps1$overlapping.genes),]
a <- overlaps2[grepl('NCK1', overlaps2$overlapping.genes) & !is.na(overlaps2$overlapping.genes),]
nck1 <- plotDMR_comparison(a, b, 
                           beta_buccal, beta_cervical,
                           pheno_buccal, pheno_cervical)

b <- overlaps1[grepl('CCDC88C', overlaps1$overlapping.genes) & !is.na(overlaps1$overlapping.genes),]
a <- overlaps2[grepl('CCDC88C', overlaps2$overlapping.genes) & !is.na(overlaps2$overlapping.genes),]
ccd <- plotDMR_comparison(a, b, 
                           beta_buccal, beta_cervical,
                           pheno_buccal, pheno_cervical)

# Supplementary table for overlaps
overlaps <- as.data.frame(overlaps1) |> 
  dplyr::select(seqnames, start, end, width, strand, no.cpgs, overlapping.genes, meandiff) |> 
  dplyr::rename(meandiff_buccal = meandiff)

overlaps$meandiff_cervical <- overlaps2$meandiff

# Objects are saved as pdfs separately to be compiled in illustrator.
ggsave(file.path(sepPath,"1c.pdf"), units = "mm", width = 44, height = 77, nck1)
ggsave(file.path(sepPath,"1d.pdf"), units = "mm", width = 44, height = 77, rp11)
ggsave(file.path(sepPath,"1e.pdf"), units = "mm", width = 44, height = 77, ccd)
ggsave(file.path(sepPath,"1f.pdf"), units = "mm", width = 44, height = 77, ltbp4)

```


## Compile and save

```{r, message=FALSE, echo=FALSE}

layout <- "
AAAA
AAAA
BBBB
BBBB
CDEF
"


pdf_width <- 8.3  # Width in inches (a4)
pdf_height <- 11.7   # Height in inches (a4: 11.7)


compiled <- a1 + b1 + nck1  + rp11 + ccd + ltbp4 
  
compiled <-  compiled + plot_annotation(tag_levels = 'a') + 
  plot_layout(design = layout)

ggsave(file.path(outPath,"F1.pdf"), width = pdf_width, height = pdf_height, compiled)
ggsave(file.path(jpgPath,"F1.jpg"), width = pdf_width, height = pdf_height, compiled)
```



# Extended Data 6: Buccal: GO analysis of DMPs

```{r}
load(here("1-ewas","1-buccal","2-output","gsea-clusterProfiler.Rdata"))

## Visualize -------------------
bp_dot <- enrichplot::dotplot(bp) 
#mf_dot <- enrichplot::dotplot(mf) 
#cc_dot <- enrichplot::dotplot(cc) 

layout <- "
A
"


pdf_width <- 8.3  # Width in inches (a4)
pdf_height <- 5   # Height in inches (a4: 11.7)


compiled <- bp_dot 
  
compiled <-  compiled  + 
  plot_layout(design = layout)&
    theme(plot.tag = element_text(face = 'bold'))

ggsave(file.path(outPath,"ED7.pdf"), width = pdf_width, height = pdf_height, compiled)
# ggsave(file.path(jpgPath,"ED7.jpg"), width = pdf_width, height = pdf_height, compiled)
```


# Extended Data 7: Cervical: GO analysis of DMPs

```{r}
load(here("1-ewas","2-cervical","2-output","gsea-clusterProfiler.Rdata"))

## Visualize -------------------
bp_dot <- enrichplot::dotplot(bp) 
mf_dot <- enrichplot::dotplot(mf) 
cc_dot <- enrichplot::dotplot(cc) 

layout <- "
A
B
C
"


pdf_width <- 8.3  # Width in inches (a4)
pdf_height <- 11.7   # Height in inches (a4: 11.7)


compiled <- bp_dot + mf_dot + cc_dot 
  
compiled <-  compiled + plot_annotation(tag_levels = 'a') + 
  plot_layout(design = layout)&
    theme(plot.tag = element_text(face = 'bold'))

ggsave(file.path(outPath,"ED8.pdf"), width = pdf_width, height = pdf_height, compiled)
# ggsave(file.path(jpgPath,"ED8.jpg"), width = pdf_width, height = pdf_height, compiled)
```

# Extended Data 8: Index training and evaluation 

## a) Buccal: Internal validation AUC 

```{r Internal Validation AUC plot1, echo=FALSE, message=FALSE}
 
ED1a <- plot_performance_compare(training_buccal, training_buccal$AUC_val, sets = 12) + labs(title = 'WID-buccal') + annotate("segment", x = 25000, xend = 30000, y = 1, yend = 0.93, colour = colour.green, size=1, alpha=0.6, arrow=arrow())
ED1a
```

## b) Buccal: OOB 

```{r}
ED1b <-plot_performance_compare(training_buccal, training_buccal$AUC_oob, sets = 12, legendPos="")+ annotate("segment", x = 25000, xend = 30000, y = 1, yend = 0.89, colour = colour.green, size=1, alpha=0.6, arrow=arrow())

ED1b 
```


## c) Cervical: Internal validation AUC 

```{r Internal Validation AUC plot2, echo=FALSE, message=FALSE}
 
ED1c <- plot_performance_compare(training_cervical, training_cervical$AUC_val, sets = 12,legendPos="") + labs(title = 'WID-cervical') + annotate("segment", x = 25000, xend = 30000, y = 1, yend = 0.93, colour = colour.blue, size=1, alpha=0.6, arrow=arrow())
ED1c
```

## d) Cervical: OOB 

```{r}
ED1d <-plot_performance_compare(training_cervical, training_cervical$AUC_oob, sets = 12,legendPos="")  + annotate("segment", x = 25000, xend = 30000, y = 0.93, yend = 0.82, colour = colour.blue, size=1, alpha=0.6, arrow=arrow())

ED1d 
```
## e) Blood: Internal validation AUC 

```{r Internal Validation AUC plot3, echo=FALSE, message=FALSE}
 
ED1e <- plot_performance_compare(training_blood, training_blood$AUC_val, sets = 12) + labs(title = 'WID-blood') + annotate("segment", x = 25000, xend = 30000, y = 0.9, yend = 0.98, colour = colour.red, size=1, alpha=0.6, arrow=arrow())
ED1e
```

## f) Blood: OOB 

```{r}
ED1f <-plot_performance_compare(training_blood, training_blood$AUC_oob, sets = 12,legendPos="")  + annotate("segment", x = 25000, xend = 30000, y = 1, yend = 0.91, colour = colour.red, size=1, alpha=0.6, arrow=arrow())

ED1f
```




## Compile and save

```{r, message=FALSE, echo=FALSE}

layout <- "
AA
BC
BC
DE
DE
FG
FG
"


pdf_width <- 8.3  # Width in inches (a4)
pdf_height <- 11.7   # Height in inches (a4: 11.7)


compiled <- guide_area() + ED1a + ED1b + ED1c + ED1d + ED1e + ED1f
  
compiled <-  compiled + plot_annotation(tag_levels = 'a') + 
  plot_layout(design = layout, guides='collect',axes = "collect")

ggsave(file.path(outPath,"ED2.pdf"), width = pdf_width, height = pdf_height, compiled)
ggsave(file.path(jpgPath,"ED2.jpg"), width = pdf_width, height = pdf_height, compiled)
```






# Extended Data 9: AUC of unadjusted indices, stratified by ic and menopause

## a) Buccal: IC stratification

```{r}

s7a <- plot_roc_dichot(factor(buccal_val$type), buccal_val$WID_buccal_BC, factor(buccal_val$ic_stratification), col1 = "black", col2 = colour.blue, col3 = colour.green, title1 = "All", title2 = paste("IC>", round(median(buccal_val$ic), 2), sep = ""), title3 = paste("IC<", round(median(buccal_val$ic), 2), sep = ""), style = "default",
                            direction = "<",textSize=textSize)  +
  ggtitle("<span style='font-size: 15pt;'>WID-buccal-BC</font>") +
  theme(plot.title = element_markdown())

s7a
```

## b) Buccal: menopause stratification

```{r}

s7b <- plot_roc_dichot(factor(buccal_val$type), buccal_val$WID_buccal_BC, factor(buccal_val$menopause), col1 = "black", col2 = colour.blue, col3 = colour.green, title1 = "All", title2 = "Post", title3 = "Pre", style = "default",
                            direction = "<",textSize=textSize)

s7b
```


## c) buccal: age at menarche

```{r}

temp <- buccal_val[which(!is.na(buccal_val$age_at_menarche)),]

temp$menarche_stratification <- temp$age_at_menarche <= median(temp$age_at_menarche)

s7c <- plot_roc_dichot(factor(temp$type), temp$WID_buccal_BC_adj, factor(temp$menarche_stratification), col1 = "black", col2 = colour.blue, col3 = colour.green, title1 = "All", title2 = paste("Menarche>", round(median(temp$age_at_menarche), 2), sep = ""), title3 = paste("Menarche<", round(median(temp$age_at_menarche), 2), sep = ""), style = "default",
                      direction = "<",textSize=textSize)

s7c

# Statistics
# D = 2.5038, df = 74.845, p-value = 0.01446
roc.menarche_low <- roc(temp$type[temp$menarche_stratification==T], temp$WID_buccal_BC_adj[temp$menarche_stratification==T], direction = "<")
roc.menarche_high <- roc(temp$type[temp$menarche_stratification==F], temp$WID_buccal_BC_adj[temp$menarche_stratification==F], direction = "<")
roc.test(roc.menarche_low,roc.menarche_high, are.paired=F)
```



## d) Cervical: IC stratification

```{r}

s7d <- plot_roc_dichot(factor(cervical_val$type), cervical_val$WID_cervical_BC, factor(cervical_val$ic_stratification), col1 = "black", col2 = colour.blue, col3 = colour.green, title1 = "All", title2 = paste("IC>", round(median(cervical_val$ic), 2), sep = ""), title3 = paste("IC<", round(median(cervical_val$ic), 2), sep = ""), style = "default",
                            direction = "<",textSize=textSize) +
  ggtitle("<span style='font-size: 15pt;'>WID-cervical-BC</font>") +
  theme(plot.title = element_markdown())


s7d
```
## e) Cervical: menopause stratification

```{r}

s7e <- plot_roc_dichot(factor(cervical_val$type), cervical_val$WID_cervical_BC, factor(cervical_val$menopause), col1 = "black", col2 = colour.blue, col3 = colour.green, title1 = "All", title2 = "Post", title3 = "Pre", style = "default",
                            direction = "<",textSize=textSize)

s7e
```

## f) Cervical: age at menarche 

```{r}
temp <- cervical_val[which(!is.na(cervical_val$age_at_menarche)),]

temp$menarche_stratification <- temp$age_at_menarche <= median(temp$age_at_menarche)


s7f <- plot_roc_dichot(factor(temp$type), temp$WID_cervical_BC_adj, factor(temp$menarche_stratification), col1 = "black", col2 = colour.blue, col3 = colour.green, title1 = "All", title2 = paste("Menarche>", round(median(temp$age_at_menarche), 2), sep = ""), title3 = paste("Menarche<", round(median(temp$age_at_menarche), 2), sep = ""), style = "default",
                      direction = "<",textSize=textSize)

s7f

# Statistics
# D = -1.0473, df = 97.068, p-value = 0.2976
roc.menarche_low <- roc(temp$type[temp$menarche_stratification==T], temp$WID_cervical_BC_adj[temp$menarche_stratification==T], direction = "<")
roc.menarche_high <- roc(temp$type[temp$menarche_stratification==F], temp$WID_cervical_BC_adj[temp$menarche_stratification==F], direction = "<")
roc.test(roc.menarche_low,roc.menarche_high, are.paired=F)

s7f
```




## g) Blood: Neutrophil stratification

```{r}

s7g <- plot_roc_dichot(factor(blood_val$type), blood_val$WID_blood_BC, factor(blood_val$neutro_stratification), col1 = "black", col2 = colour.blue, col3 = colour.green, title1 = "All", title2 = paste("Neutrophil>", round(median(blood_val$neutro_stratification), 2), sep = ""), title3 = paste("Neutrophil<", round(median(blood_val$neutro_stratification), 2), sep = ""), style = "default",
                            direction = "<",textSize=textSize) +
  ggtitle("<span style='font-size: 15pt;'>WID-blood-BC</font>") +
  theme(plot.title = element_markdown())


s7g
```

## Compile and save

```{r, message=FALSE, echo=FALSE}

layout <- "
ABC
DEF
G##
"


pdf_width <- 8.3  # Width in inches (a4)
pdf_height <- 8   # Height in inches (a4: 11.7)


compiled <- s7a + s7b + s7c + s7d + s7e + s7f + s7g
  
compiled <-  compiled + plot_annotation(tag_levels = 'a') + 
  plot_layout(design = layout)&
    theme(plot.tag = element_text(face = 'bold'))

ggsave(file.path(outPath,"ED9.pdf"), width = pdf_width, height = pdf_height, compiled)
# ggsave(file.path(jpgPath,"ED9.jpg"), width = pdf_width, height = pdf_height, compiled)
```


# Extended Data 10: Association of the WID-BC-buccal and WID-BC-cervical index with clinical and epidemiological parameters.

All that are NS are not included for visability of the figure!


## a) Age at menarche (years)

Not included: 
+ 
 geom_signif(comparisons = list(c("7-11", "12-13"), c("7-11", ">13"), c("12-13", ">13")),
              map_signif_level = TRUE,
              textsize = 3, vjust = -0.3)

```{r}
# age at menarche (years) 7-11, 12-13, >13
# Define the breaks for the age categories
temp <- buccal_val[which(!is.na(buccal_val$age_at_menarche)),]

breaks <- c(7, 11, 13, Inf)  # Inf represents values greater than 13

# Create the new factor column
temp$age_at_menarche_factor <- cut(temp$age_at_menarche, breaks = breaks, 
                                              labels = c("7-11", "12-13", ">13"), include.lowest = TRUE)

# Convert the new column to a factor
temp$age_at_menarche_factor <- as.factor(temp$age_at_menarche_factor)





s8a <- ggplot(data=temp, aes(x = age_at_menarche_factor, y = WID_buccal_BC_adj)) +
  geom_boxplot(aes(fill = type),
               alpha = 0.2, outlier.shape = NA) +
  ggbeeswarm::geom_quasirandom(aes(colour = type),
                               alpha = 0.5, size = 0.5) +
  scale_color_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control)) +  # Add linear fit with confidence interval
  labs(
    x = "Age at menarche (years)",
    y = "WID-buccal-BC index (adj.)",
    title = "WID-buccal-BC"
  ) +
  theme_minimal() +
  theme(legend.title = element_blank()) +
  theme(legend.position = "none") +  # Remove legend
  theme(panel.background = element_blank())+
  facet_wrap(~type) +  # Facet by 'type'
  scale_fill_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control))


print(s8a)
```

## b) first degree relatives with BC 0 and ≥1

Not included: 
+ 
  geom_signif(comparisons = list(c("0","1")),
              map_signif_level = TRUE,
              textsize = 3, vjust = -0.3)

```{r}
# first degree relatives with BC 0 and ≥1
s8b <- ggplot(data=temp, aes(x = as.factor(first_degree_relatives_with_BC),
                     y = WID_buccal_BC_adj)) +
  geom_boxplot(aes(fill = type),
               alpha = 0.2, outlier.shape = NA) +
  ggbeeswarm::geom_quasirandom(aes(colour = type),
                               alpha = 0.5, size = 0.5) +
  scale_color_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control)) +  # Add linear fit with confidence interval
  labs(
    x = "Number of first degree relatives \n with BC",
    y = "WID-buccal-BC index (adj.)"
  ) +
  theme_minimal() +
  theme(legend.title = element_blank()) +
  theme(legend.position = "none") +  # Remove legend
  theme(panel.background = element_blank())+
  facet_wrap(~type) +  # Facet by 'type'
#  ggpubr::stat_compare_means(comparisons = list(c("0","1")), label = 'p.signif', label.y = c(2.5), vjust = +0.1,textsize=2, #hide.ns = F) +
  scale_fill_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control))


s8b
```

## c) age at first live birth (years) <20, 20-24, 25-29, ≥30
Not included:

+ 
 geom_signif(comparisons = list(
  c("<20", "20-24"),
  c("<20", "≥30"),
  c("20-24", "≥30")
),
              map_signif_level = TRUE,
              textsize = 3, vjust = -0.3)
              
```{r}

temp <- buccal_val[which(!is.na(buccal_val$Pregnancy_age_1st_live_born)),]

# Define the breaks and labels 
breaks <- c(0, 20, 24, 29, Inf)
labels <- c("<20", "20-24", "25-29", "≥30")

# Create the new factor column
temp$Pregnancy_age_1st_live_born_factor <- cut(
  temp$Pregnancy_age_1st_live_born,
  breaks = breaks,
  labels = labels,
  right = FALSE  # Include the left endpoint in the interval
)

# Convert to factor
temp$Pregnancy_age_1st_live_born_factor <- as.factor(temp$Pregnancy_age_1st_live_born_factor)




s8c <- ggplot(data=temp, aes(x = Pregnancy_age_1st_live_born_factor, y = WID_buccal_BC_adj)) +
  geom_boxplot(aes(fill = type),
               alpha = 0.2, outlier.shape = NA) +
  ggbeeswarm::geom_quasirandom(aes(colour = type),
                               alpha = 0.5, size = 0.5) +
  scale_color_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control)) +  # Add linear fit with confidence interval
  labs(
    x = "Age at first live birth (years)",
    y = "WID-buccal-BC index (adj.)"
  ) +
  theme_minimal() +
  theme(legend.title = element_blank(),axis.text.x = element_text(angle = 45, hjust = 1)) +
  theme(legend.position = "none") +  # Remove legend
  theme(panel.background = element_blank())+
  facet_wrap(~type) +  # Facet by 'type' 
 # ggpubr::stat_compare_means(comparisons = list(
 # c("<20", "20-24"),
#  c("<20", "≥30"),
 # c("20-24", "≥30")), label = 'p.signif', label.y = c(2.5), vjust = +0.1,textsize=2, hide.ns = F) +
  scale_fill_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control))


s8c







```

## d) menopause, pre, post

Not included: 

 + 
  geom_signif(comparisons = list(c("Pre","Post")),
              map_signif_level = TRUE,
              textsize = 3, vjust = -0.3)

```{r}

buccal_val$menopause <- factor(buccal_val$menopause, levels=c("Pre","Post"))

s8d <- ggplot(data=temp, aes(x = menopause, y = WID_buccal_BC_adj)) +
  geom_boxplot(aes(fill = type),
               alpha = 0.2, outlier.shape = NA) +
  ggbeeswarm::geom_quasirandom(aes(colour = type),
                               alpha = 0.5, size = 0.5) +
  scale_color_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control)) +  # Add linear fit with confidence interval
  labs(
    x = "Menopausal status",
    y = "WID-buccal-BC index (adj.)"
  ) +
  theme_minimal() +
  theme(legend.title = element_blank()) +
  theme(legend.position = "none") +  # Remove legend
  theme(panel.background = element_blank())+
  facet_wrap(~type) +  # Facet by 'type' 
# ggpubr::stat_compare_means(comparisons = list(
# c("Pre","Post")), label = 'p.signif', label.y = c(2.5), vjust = +0.1,textsize=2, hide.ns = F) +
  scale_fill_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control))

s8d
```

## e) hrt ever no, yes
Post-menopausal women only

not included: + 
  geom_signif(comparisons = list(c("No","Yes")),
              map_signif_level = TRUE,
              textsize = 3, vjust = -0.3)
              

```{r}

filtered_tmp <- subset(buccal_val, menopause == "Post")

s8e <- ggplot(data=filtered_tmp, aes(x = HRT_ever, y = WID_buccal_BC_adj)) +
  geom_boxplot(aes(fill = type),
               alpha = 0.2, outlier.shape = NA) +
  ggbeeswarm::geom_quasirandom(aes(colour = type),
                               alpha = 0.5, size = 0.5) +
  scale_color_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control)) +  # Add linear fit with confidence interval
  labs(
    x = "HRT ever",
    y = "WID-buccal-BC index (adj.)"
  ) +
  theme_minimal() +
  theme(legend.title = element_blank()) +
  theme(legend.position = "none") +  # Remove legend
  theme(panel.background = element_blank()) +
  facet_wrap(~type) +
# ggpubr::stat_compare_means(comparisons = list(
# c("No","Yes")), label = 'p.signif', label.y = c(2.5), vjust = +0.1,textsize=2, hide.ns = F) +
  scale_fill_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control)) 
s8e
```


## f) nodal status, n0,n1n2,n3
all comparisons made: list(
  c("N0", "N1"),
  c("N1", "N2"),
  c("N2", "N3"),
  c("N0", "N2"),
  c("N0", "N3"),
  c("N1", "N3"))
  
only the significance retained for illustration purposes:
list(
  c("N1", "N2"),
  c("N2", "N3"),
  c("N0", "N2"))


```{r}

temp <- buccal_val[which(buccal_val$stage_N!=""),]

s8f <- ggplot(data=temp, aes(x = stage_N, y = WID_buccal_BC_adj)) +
  geom_boxplot(aes(fill = type),
               alpha = 0.2, outlier.shape = NA) +
  ggbeeswarm::geom_quasirandom(aes(colour = type),
                               alpha = 0.5, size = 0.5) +
  scale_color_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control)) +  # Add linear fit with confidence interval
  labs(
    x = "Nodal status",
    y = "WID-buccal-BC index (adj.)"
  ) +
  theme_minimal() +
  theme(legend.title = element_blank()) +
  theme(legend.position = "none") +  # Remove legend
  theme(panel.background = element_blank()) +
  ggpubr::stat_compare_means(comparisons = list(
  c("N1", "N2"),
  c("N2", "N3"),
  c("N0", "N2")), label = 'p.signif', vjust = +0.2, label.y = c(2.3,2.4,2.6)) +
  scale_fill_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control))

s8f 


```






## g) Age at menarche (years)

Not included: 
+ 
 geom_signif(comparisons = list(c("7-11", "12-13"), c("7-11", ">13"), c("12-13", ">13")),
              map_signif_level = TRUE,
              textsize = 3, vjust = -0.3)

```{r}
# age at menarche (years) 7-11, 12-13, >13
# Define the breaks for the age categories
temp <- cervical_val[which(!is.na(cervical_val$age_at_menarche)),]

breaks <- c(7, 11, 13, Inf)  # Inf represents values greater than 13

# Create the new factor column
temp$age_at_menarche_factor <- cut(temp$age_at_menarche, breaks = breaks, 
                                              labels = c("7-11", "12-13", ">13"), include.lowest = TRUE)

# Convert the new column to a factor
temp$age_at_menarche_factor <- as.factor(temp$age_at_menarche_factor)





s9a <- ggplot(data=temp, aes(x = age_at_menarche_factor, y = WID_cervical_BC_adj)) +
  geom_boxplot(aes(fill = type),
               alpha = 0.2, outlier.shape = NA) +
  ggbeeswarm::geom_quasirandom(aes(colour = type),
                               alpha = 0.5, size = 0.5) +
  scale_color_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control)) +  # Add linear fit with confidence interval
  labs(
    x = "Age at menarche (years)",
    y = "WID-cervical-BC index (adj.)",
    title = "WID-cervical-BC"
  ) +
  theme_minimal() +
  theme(legend.title = element_blank()) +
  theme(legend.position = "none") +  # Remove legend
  theme(panel.background = element_blank())+
  facet_wrap(~type) +  # Facet by 'type'
# ggpubr::stat_compare_means(comparisons = list(c("7-11", "12-13"), c("7-11", ">13"), c("12-13", ">13")), label = 'p.signif', #label.y = c(2.5), vjust = +0.1,textsize=2, hide.ns = F) +
  scale_fill_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control))


print(s9a)
```

## h) first degree relatives with BC 0 and ≥1

Not included: 
+ 
  geom_signif(comparisons = list(c("0","1")),
              map_signif_level = TRUE,
              textsize = 3, vjust = -0.3)

```{r}
# first degree relatives with BC 0 and ≥1
s9b <- ggplot(data=cervical_val, aes(x = as.factor(first_degree_relatives_with_BC),
                     y = WID_cervical_BC_adj)) +
  geom_boxplot(aes(fill = type),
               alpha = 0.2, outlier.shape = NA) +
  ggbeeswarm::geom_quasirandom(aes(colour = type),
                               alpha = 0.5, size = 0.5) +
  scale_color_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control)) +  # Add linear fit with confidence interval
  labs(
    x = "Number of first degree relatives \n with BC",
    y = "WID-cervical-BC index (adj.)"
  ) +
  theme_minimal() +
  theme(legend.title = element_blank()) +
  theme(legend.position = "none") +  # Remove legend
  theme(panel.background = element_blank())+
  facet_wrap(~type) +  # Facet by 'type'
#  ggpubr::stat_compare_means(comparisons = list(c("0","1")), label = 'p.signif', label.y = c(2.5), vjust = +0.1,textsize=2, #hide.ns = F) +
  scale_fill_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control))


s9b
```

## i) age at first live birth (years) <20, 20-24, 25-29, ≥30
Not included:

+ 
 geom_signif(comparisons = list(
  c("<20", "20-24"),
  c("<20", "≥30"),
  c("20-24", "≥30")
),
              map_signif_level = TRUE,
              textsize = 3, vjust = -0.3)
```{r}

temp <- cervical_val[which(!is.na(cervical_val$Pregnancy_age_1st_live_born)),]

# Define the breaks and labels 
breaks <- c(0, 20, 24, 29, Inf)
labels <- c("<20", "20-24", "25-29", "≥30")

# Create the new factor column
temp$Pregnancy_age_1st_live_born_factor <- cut(
  temp$Pregnancy_age_1st_live_born,
  breaks = breaks,
  labels = labels,
  right = FALSE  # Include the left endpoint in the interval
)

# Convert to factor
temp$Pregnancy_age_1st_live_born_factor <- as.factor(temp$Pregnancy_age_1st_live_born_factor)




s9c <- ggplot(data=temp, aes(x = Pregnancy_age_1st_live_born_factor, y = WID_cervical_BC_adj)) +
  geom_boxplot(aes(fill = type),
               alpha = 0.2, outlier.shape = NA) +
  ggbeeswarm::geom_quasirandom(aes(colour = type),
                               alpha = 0.5, size = 0.5) +
  scale_color_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control)) +  # Add linear fit with confidence interval
  labs(
    x = "Age at first live birth (years)",
    y = "WID-cervical-BC index (adj.)"
  ) +
  theme_minimal() +
  theme(legend.title = element_blank(),axis.text.x = element_text(angle = 45, hjust = 1)) +
  theme(legend.position = "none") +  # Remove legend
  theme(panel.background = element_blank())+
  facet_wrap(~type) +  # Facet by 'type' 
# ggpubr::stat_compare_means(comparisons = list(
# c("<20", "20-24"),
#  c("<20", "≥30"),
# c("20-24", "≥30")), label = 'p.signif', label.y = c(2.5), vjust = +0.1,textsize=2, hide.ns = F) +
  scale_fill_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control))


s9c







```

## j) menopause, pre, post

Not included: 

 + 
  geom_signif(comparisons = list(c("Pre","Post")),
              map_signif_level = TRUE,
              textsize = 3, vjust = -0.3)

```{r}

cervical_val$menopause <- factor(cervical_val$menopause, levels=c("Pre","Post"))

s9d <- ggplot(data=cervical_val, aes(x = menopause, y = WID_cervical_BC_adj)) +
  geom_boxplot(aes(fill = type),
               alpha = 0.2, outlier.shape = NA) +
  ggbeeswarm::geom_quasirandom(aes(colour = type),
                               alpha = 0.5, size = 0.5) +
  scale_color_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control)) +  # Add linear fit with confidence interval
  labs(
    x = "Menopausal status",
    y = "WID-cervical-BC index (adj.)"
  ) +
  theme_minimal() +
  theme(legend.title = element_blank()) +
  theme(legend.position = "none") +  # Remove legend
  theme(panel.background = element_blank())+
  facet_wrap(~type) +  # Facet by 'type' 
# ggpubr::stat_compare_means(comparisons = list(
# c("Pre","Post")), label = 'p.signif', label.y = c(2.5), vjust = +0.1,textsize=2, hide.ns = F) +
  scale_fill_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control))

s9d
```

## k) hrt ever no, yes
Post-menopausal women only

not included: + 
  geom_signif(comparisons = list(c("No","Yes")),
              map_signif_level = TRUE,
              textsize = 3, vjust = -0.3)
              

```{r}
filtered_tmp <- subset(cervical_val, menopause == "Post")

s9e <- ggplot(data=filtered_tmp, aes(x = HRT_ever, y = WID_cervical_BC_adj)) +
  geom_boxplot(aes(fill = type),
               alpha = 0.2, outlier.shape = NA) +
  ggbeeswarm::geom_quasirandom(aes(colour = type),
                               alpha = 0.5, size = 0.5) +
  scale_color_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control)) +  # Add linear fit with confidence interval
  labs(
    x = "HRT ever",
    y = "WID-cervical-BC index (adj.)"
  ) +
  theme_minimal() +
  theme(legend.title = element_blank()) +
  theme(legend.position = "none") +  # Remove legend
  theme(panel.background = element_blank()) +
  facet_wrap(~type) +
# ggpubr::stat_compare_means(comparisons = list(
# c("No","Yes")), label = 'p.signif', label.y = c(2.5), vjust = +0.1,textsize=2, hide.ns = F) +
  scale_fill_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control)) 
s9e
```


## l) nodal status, n0,n1n2,n3
all comparisons made: list(
  c("N0", "N1"),
  c("N1", "N2"),
  c("N2", "N3"),
  c("N0", "N2"),
  c("N0", "N3"),
  c("N1", "N3"))
  
only the significance retained for illustration purposes:
list(
  c("N1", "N2"),
  c("N2", "N3"),
  c("N0", "N2"))


```{r}

temp <- cervical_val[which(cervical_val$stage_N!=""),]

s9f <- ggplot(data=temp, aes(x = stage_N, y = WID_cervical_BC_adj)) +
  geom_boxplot(aes(fill = type),
               alpha = 0.2, outlier.shape = NA) +
  ggbeeswarm::geom_quasirandom(aes(colour = type),
                               alpha = 0.5, size = 0.5) +
  scale_color_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control)) +  # Add linear fit with confidence interval
  labs(
    x = "Nodal status",
    y = "WID-cervical-BC index (adj.)"
  ) +
  theme_minimal() +
  theme(legend.title = element_blank()) +
  theme(legend.position = "none") +  # Remove legend
  theme(panel.background = element_blank()) +
  # ggpubr::stat_compare_means(comparisons = list(
  # c("N0", "N1"),
  # c("N1", "N2"),
  # c("N2", "N3"),
  # c("N0", "N2"),
  # c("N0", "N3"),
  # c("N1", "N3")), label = 'p.signif', vjust = +0.2) +
  scale_fill_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control))

s9f 


```


## Compile and save 

```{r}

layout <- "
ABC
DEF
GHI
JKL
"


pdf_width <- 8.3  # Width in inches (a4)
pdf_height <- 11.7   # Height in inches (a4: 11.7)


compiled <- s8a + s8b + s8c + s8d + s8e + s8f  + s9a + s9b + s9c + s9d + s9e + s9f 
  
compiled <-  compiled + plot_annotation(tag_levels = 'a') + 
  plot_layout(design = layout) &
    theme(plot.tag = element_text(face = 'bold'))

ggsave(file.path(outPath,"ED10.pdf"), width = pdf_width, height = pdf_height, compiled)
# ggsave(file.path(jpgPath,"ED10.jpg"), width = pdf_width, height = pdf_height, compiled)


```


# Extended Data 11: WID-cervical and WID-buccal in blood datasets

```{r}
## FORECEE
a <- ggplot(data = blood_3c_val, aes(x = hepidish_Neutro, y = WID_buccal_BC_adj_neutro, color = type, fill = factor(type))) +
  geom_point(alpha=0.4,size=1) +  # Plot points
  geom_smooth(method = "lm", se = TRUE, aes(group = type)) +
  scale_color_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control)) +  # Add linear fit with confidence interval
  labs(
    y = "WID-buccal-BC index\n(neutrophil adjusted)",
    x = "Neutrophil proportion"
  ) +
  theme_minimal() +
  theme(legend.title = element_blank()) +
  theme(legend.position = "none") +  # Remove legend
  theme(panel.background = element_blank()) + 
  stat_cor(method = "pearson", show.legend = FALSE, label.x.npc='left', label.y.npc='top', size = 3) + 
  scale_fill_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control))  +
  ggtitle("<span style='font-size: 15pt;'>FORECEE</font>") +
  theme(plot.title = element_markdown())




b <- plot_roc_dichot(factor(blood_3c_val$type), blood_3c_val$WID_buccal_BC_adj_neutro, factor(blood_3c_val$neutro_stratification), col1 = "black", col2 = colour.blue, col3 = colour.green, title1 = "WID-buccal-BC", title2 = paste("Neutro>", round(median(blood_3c_val$hepidish_Neutro), 2), sep = ""), title3 = paste("Neutro<", round(median(blood_3c_val$hepidish_Neutro), 2), sep = ""), style = "default",
                            direction = "<",textSize=textSize)


c <- ggplot(data = blood_3c_val, aes(x = hepidish_Neutro, y = WID_cervical_BC_adj_neutro, color = type, fill = factor(type))) +
  geom_point(alpha=0.4,size=1) +  # Plot points
  geom_smooth(method = "lm", se = TRUE, aes(group = type)) +
  scale_color_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control)) +  # Add linear fit with confidence interval
  labs(
    y = "WID-cervical-BC index\n(neutrophil adjusted)",
    x = "Neutrophil proportion"
  ) +
  theme_minimal() +
  theme(legend.title = element_blank()) +
  theme(legend.position = "none") +  # Remove legend
  theme(panel.background = element_blank()) + 
  stat_cor(method = "pearson", show.legend = FALSE, label.x.npc='left', label.y.npc='top', size = 3) + 
  scale_fill_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control)) 

d <- plot_roc_dichot(factor(blood_3c_val$type), blood_3c_val$WID_cervical_BC_adj_neutro, factor(blood_3c_val$neutro_stratification), col1 = "black", col2 = colour.blue, col3 = colour.green, title1 = "WID-cervical-BC", title2 = paste("Neutro>", round(median(blood_3c_val$hepidish_Neutro), 2), sep = ""), title3 = paste("Neutro<", round(median(blood_3c_val$hepidish_Neutro), 2), sep = ""), style = "default",
                            direction = "<",textSize=textSize)


## WANG
e <- ggplot(data = blood_val, aes(x = hepidish_Neutro, y = WID_buccal_BC_adj, color = type, fill = factor(type))) +
  geom_point(alpha=0.5,size=1) +  # Plot points
  geom_smooth(method = "lm", se = TRUE, aes(group = type)) +
  scale_color_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control)) +  # Add linear fit with confidence interval
  labs(
    y = "WID-buccal-BC index\n(neutrophil adjusted)",
    x = "Neutrophil proportion") +
  theme_minimal() +
  theme(legend.title = element_blank()) +
  theme(legend.position = "none") +  # Remove legend
  theme(panel.background = element_blank()) + 
  stat_cor(method = "pearson", show.legend = FALSE, label.x.npc='left', label.y.npc='top', size = 3) + 
  scale_fill_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control)) +
  ggtitle("<span style='font-size: 15pt;'>GSE237036\n</font>") +
  theme(plot.title = element_markdown())




f <- plot_roc_dichot(factor(blood_val$type), blood_val$WID_buccal_BC_adj, factor(blood_val$neutro_stratification), col1 = "black", col2 = colour.blue, col3 = colour.green, title1 = "WID-buccal-BC", title2 = paste("Neutro>", round(median(blood_3c_val$hepidish_Neutro), 2), sep = ""), title3 = paste("Neutro<", round(median(blood_val$hepidish_Neutro), 2), sep = ""), style = "default",
                            direction = "<",textSize=textSize) 


g <- ggplot(data = blood_val, aes(x = hepidish_Neutro, y = WID_cervical_BC_adj, color = type, fill = factor(type))) +
  geom_point(alpha=0.4,size=1) +  # Plot points
  geom_smooth(method = "lm", se = TRUE, aes(group = type)) +
  scale_color_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control)) +  # Add linear fit with confidence interval
  labs(
    y = "WID-cervical-BC index\n(neutrophil adjusted)",
    x = "Neutrophil proportion"
  ) +
  theme_minimal() +
  theme(legend.title = element_blank()) +
  theme(legend.position = "none") +  # Remove legend
  theme(panel.background = element_blank()) + 
  stat_cor(method = "pearson", show.legend = FALSE, label.x.npc='left', label.y.npc='top', size = 3) + 
  scale_fill_manual(values = c("BC case" = colour.Breast, "Control" = colour.Control)) 




h <- plot_roc_dichot(factor(blood_val$type), blood_val$WID_cervical_BC_adj, factor(blood_val$neutro_stratification), col1 = "black", col2 = colour.blue, col3 = colour.green, title1 = "WID-cervical-BC", title2 = paste("Neutro>", round(median(blood_val$hepidish_Neutro), 2), sep = ""), title3 = paste("Neutro<", round(median(blood_val$hepidish_Neutro), 2), sep = ""), style = "default",
                            direction = "<",textSize=textSize)


pdf_width <- 10  # Width in inches (a4: 8.3)
pdf_height <- 6   # Height in inches (a4: 11.7)


layout <- "
ABCD
EFGH
"
compiled <- a + b+ c+ d+ e+ f+g+h
  
compiled <-  compiled + plot_annotation(tag_levels = 'a',title = 'WID-buccal/cervical-BC in blood samples') + 
  plot_layout(design = layout) &
    theme(plot.tag = element_text(face = 'bold'))

ggsave(file.path(outPath,"ED11.pdf"), width = pdf_width, height = pdf_height, compiled)
# ggsave(file.path(jpgPath,"ED11.jpg"), width = pdf_width, height = pdf_height, compiled)



```